<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
</head>

<body>
<div class="content">
	<div class="item"  id="item1">
	<div class="options">
		<h1>Download <b>QTLtools</b></h1>
		<p>
		QTLtools releases available for download:
		</p>
		<table width=100%>
			<tr>
				<th width=20%>Version 1.3.1</th>
				<th width=80%>Description</th>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.3.1_source.tar.gz" download>v1.3.1 (Source code)</a>
				</td>
				<td>
				<b>Major bug fix regarding missing genotypes in <u>cis</u>, <u>correct</u>, <u>gwas</u>, <u>pca</u>, <u>rep</u>, <u>trans</u>, <u>rtc-union</u></b>. Multiple new modes. Other minor bug fixes and many improvements.
				</td>
			</tr>
		</table>
		
		<table width=100%>
			<tr>
				<th width=20%>Version 1.2</th>
				<th width=80%>Description</th>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.2_source.tar.gz" download>v1.2 (Source code)</a>
				</td>
				<td>
				<b>The mode <u>quan</u> is not compatible with the previous versions </b>. Updated Eigen library to version 3.3.7. Many bug fixes.
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.2_Ubuntu16.04_x86_64.tar.gz" download>v1.2 (Binary Ubuntu16.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu16.04/x86_64 using <i>g++ 5.4.0</i>, <i>glibc 2.23</i>, <i>HTSlib 1.9</i>, and <i>R 3.6.1</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.2_Ubuntu14.04_x86_64.tar.gz" download>v1.2 (Binary Ubuntu14.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu14.04/x86_64 using <i>g++ 4.8.4</i>, <i>glibc 2.19</i>, <i>HTSlib 1.9</i>, and <i>R 3.6.1</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.2_Ubuntu12.04_x86_64.tar.gz" download>v1.2 (Binary Ubuntu12.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu12.04/x86_64 using <i>g++ 4.6.3</i>, <i>glibc 2.15</i>, <i>HTSlib 1.9</i>, and <i>R 3.6.1</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.2_CentOS7.8_x86_64.tar.gz" download>v1.2 (Binary CentOS 7.8)</a>
				</td>
				<td>
				Compiled binary on CentOS7.8/x86_64 using <i>g++ 4.8.5</i>, <i>glibc 2.17</i>, <i>HTSlib 1.9</i>, and <i>R 3.6.1</i>
				</td>
			</tr>
		</table>
		
		<table width=100%>
			<tr>
				<th width=20%>Version 1.1</th>
				<th width=80%>Description</th>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.1_source.tar.gz" download>v1.1 (Source code)</a>
				</td>
				<td>
				The mode <b>match</b> is now called <b>mbv</b>. Lots of bugs were fixed the following modes: <b>quan</b>, <b>trans</b>, <b>fenrich</b> & <b>fdensity</b>.
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.1_Ubuntu16.04_x86_64.tar.gz" download>v1.1 (Binary Ubuntu16.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu16.04/x86_64 using <i>g++ 5.4.0</i> and <i>glibc 2.23</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.1_Ubuntu14.04_x86_64.tar.gz" download>v1.1 (Binary Ubuntu14.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu14.04/x86_64 using <i>g++ 4.8.4</i> and <i>glibc 2.19</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.1_Ubuntu12.04_x86_64.tar.gz" download>v1.1 (Binary Ubuntu12.04)</a>
				</td>
				<td>
				Compiled binary on Ubuntu12.04/x86_64 using <i>g++ 4.6.3</i> and <i>glibc 2.15</i>
				</td>
			</tr>
		</table>
		
		<br>
				
		<table width=100%>
			<tr>
				<th width=20%>Version 1.0</th>
				<th width=80%>Description</th>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_source.tar.gz" download>v1.0 (Source code)</a>
				</td>
				<td>
				This is the first stable version of the code.
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_Ubuntu16.04_x86_64.tar.gz" download>v1.0 (Binary Ubuntu16.04)</a>
				</td>
				<td>
				Compiled binary of the first stable version of the code on Ubuntu16.04/x86_64 using <i>g++ 5.4.0</i> and <i>glibc 2.23</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_Ubuntu14.04_x86_64.tar.gz" download>v1.0 (Binary Ubuntu14.04)</a>
				</td>
				<td>
				Compiled binary of the first stable version of the code on Ubuntu14.04/x86_64 using <i>g++ 4.8.4</i> and <i>glibc 2.19</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_Ubuntu12.04_x86_64.tar.gz" download>v1.0 (Binary Ubuntu12.04)</a>
				</td>
				<td>
				Compiled binary of the first stable version of the code on Ubuntu12.04/x86_64 using <i>g++ 4.6.3</i> and <i>glibc 2.15</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_CentOS6.8_x86_64.tar.gz" download>v1.0 (Binary CentOS6.8)</a>
				</td>
				<td>
				Compiled binary of the first stable version of the code on CentOS6.8/x86_64 using <i> g++ 4.7.2</i> and <i>glibc 2.12</i>
				</td>
			</tr>
			<tr>
				<td>
				<a href="https://qtltools.github.io/qtltools/binaries/QTLtools_1.0_MacOS10.11.5_x86_64.gz " download>v1.0 (Binary MacOSx10.11.5)</a>
				</td>
				<td>
				Compiled binary of the first stable version of the code on MacOSX 10.11.5 (El Capitan) using <i> Apple LLVM version 7.3.0 (clang-703.0.31)</i>
				</td>
			</tr>
		</table>
	</div>
	</div>
	
	<div class="item"  id="item1">
		<h1>Important changes in version 1.3.1</h1>
		<ul>
    <li>Major bug fix in reading missing genotypes in VCF/BCF files. This bug affected variants with a DS field in their genotype's FORMAT and have a missing genotype (DS field is .) in one of the samples, in which case genotypes for all the samples are set to missing, effectively removing this variant from the analyses. Affected modes: <u>cis</u>, <u>correct</u>, <u>gwas</u>, <u>pca</u>, <u>rep</u>, <u>trans</u>, <u>rtc-union</u>. <b>IT'S HIGHLY RECOMMENDED THAT YOU CHECK IF YOU ARE AFFECTED BY THIS, AND UPDATE TO THE LATEST VERSION IF YOU WANT TO FIX THE ISSUE.</b></li>
    <li>Multiple new modes are officially released and documentation is made available, most importantly you can now run allele specific allele (ASE) analysis.</li>
    <li>We now provide the R-squared in mode <u>cis</u>, and an option to print out the standard error. </li>
    <li>Multiple bug fixes in the <u>pca</u> mode regarding ploidy, and we remove the MT, X, and Y chromosomes by default. </li>
    <li>All documentation is updated.</li>
    <li>Changes to some option defaults.</li>
    <li>Minor bug fixes and spelling corrections.</li>
	<li>We will only release the source code from this version onwards.</li>
    </ul>
	</div>
	
	<div class="item"  id="item1">
		<h1>Important changes in version 1.2</h1>
		<p>For best performance it is highly recommended that you compile from source rather than using the precompiled binaries.</p>
		<p>
		The quan mode in version 1.2 and above is not compatible with the quantifications generated by the previous versions. This due to bug fixes and slight adjustments to the way we quantify. <b>DO NOT MIX QUANTIFICATIONS GENERATED BY EARLIER VERSIONS OF QTLTOOLS WITH QUANTIFICATIONS FROM VERSION 1.2 AND ABOVE AS THIS WILL CREATE A BIAS IN YOUR DATASET.</b>
		</p>
		
		<p>
			The Eigen library is upgraded to version 3.3.7. We have observed that this results in the principal components produced by the <b>pca</b> mode to be ever so slightly different from the previous versions. This could also impact any analysis where the --cov option is used, however in our internal testing we have not encountered this issue. <b>Thus for ongoing analyses it is recommended that you do not upgrade your QTLtools version.</b> 
		</p>
		
	</div>

	<div class="item"  id="item1">
		<h1>Testing if <b>QTLtools</b> runs on your machine</h1>
		<p>
		To test if QTLtools can run on your plateform, run:
		</p>

		<code>
		QTLtools --help
		</code>

		<p>
		If it does not complain for missing libraries, you're good to go. Otherwise, first check your <i>glibc</i> version by running:
		</p>

		<code>
		locate libc.so.6
		</code>

		<p>
		This is going to tell you where is the library. Assuming it is in <i>/lib/x86_64-linux-gnu/</i>, then check <i>glibc</i> version by running:
		</p>

		<code>
		/lib/x86_64-linux-gnu/libc.so.6
		</code>
		
		<p>
		...which gives:
		</p>

		<code>
		GNU C Library (Ubuntu EGLIBC 2.19-0ubuntu6.9) stable release version <b>2.19</b>, by Roland McGrath et al.<br>
		Copyright (C) 2014 Free Software Foundation, Inc.<br>
		This is free software; see the source for copying conditions.<br>
		...
		</code>

		<p>
		Given this, you can then choose the appropriate binary by picking up one based on an older or identical <i>glibc</i> version.
		If your version of <i>glibc</i> is older than <i>2.12</i>, it is very likely that you need to recompile QTLtools following the instructions below:
		</p>
	</div>

	<div class="item"  id="item1">
		<h1>Compiling <b>QTLtools</b></h1>
		<p style="font-size:1.4em;"><b><a href="basic_compilation.html" target='content'>Please also see Compilation examples</a></b></p>
		<p>
		If the provided binaries do not directly work, QTLtools needs to be re-compiled on your platform. For this, you may need to install few standard libraries required by QTLtools if you do not have them already installed. To do so, you need to update the path of the required libraries in the <i>Makefile</i>. 
		</p>

		<br>
		<h3>Prerequisites</h3>
		<p>The development versions of following libraries are required to compile QTLtools (you will also need the static versions of these libraries if you want to create a statically linked binary):</p>
		<ul>
			<li>zlib</li>
			<li>bzip2</li>
			<li>lzma</li>
			<li>curl</li>
			<li>gsl</li>
			<li>boost</li>
			<li>Rmath</li>
			<li>HTSlib</li>
		</ul>
		<h3>0. C++ compiler</h3>
		<p>
		Please make sure that the version of your <i>g++</i> is above 4.6. To check this, use:
		</p>

		<code>
		g++ --version
		</code>

		<br>
		<h3>1. The GNU Scientific Library</h3>
		<p>
		This library is usually installed by default on most linux platforms, otherwise instructions to install it can be found <a href="http://www.gnu.org/software/gsl/" target='_top'>here</a>. This librry does not need any modification of the <i>Makefile</i> since it is dynamically linked.
		</p>

		<br>
		<h3>2. The Boost C++ library</h3>
		<p>
		This collection of C++ libraries is also usually installed by default on most computers. Detailed instructions to install it can be found <a href="http://www.boost.org/" target='_top'>here</a>. Once installed, you need to edit the <i>Makefile</i> and set the two following variables <i>BOOST_INC</i> and <i>BOOST_LIB</i> correctly.
		</p>

		<p>
		First, find the file <i>program_options.hpp</i> where you installed the Boost libraries. For this, you can use the <i>find</i> command. Assuming that it is located here: <i>/usr/include/boost/program_options.hpp</i>, you can then set <i>BOOST_INC</i> in the <i>Makefile</i> as follows:
		</p>

		<code>
		BOOST_INC=/sur/include
		</code>
		
		<p>
		Here, do not forget to remove <i>boost/program_options.hpp</i> from the full path.

		<p>
		Then, look for the file <i>libboost_program_options.a</i>. If it is located here: <i>/usr/lib/x86_64-linux-gnu/</i>, set <i>BOOST_LIB</i> as follows:
		</p>

		<code> 
		BOOST_LIB=/usr/lib/x86_64-linux-gnu
		</code>

		<br>
		<h3>3. The standalone Rmath library</h3>
		<p>
		This library provided by R contains all basic functions to simulate from standard probability distributions (ex: qnorm, pnorm, rnorm, etc ...). Instructions to install it can be found <a href="http://cran.r-project.org/doc/manuals/r-release/R-admin.html#The-standalone-Rmath-library" target='_top'>here</a>. A pre-compiled version can be directly obtained by installing the package <i>r-mathlib</i> on Debian/Ubuntu or <i>libRmath-dev</i> on Fedora/RedHat. Otherwise, proceed as follows:</p>

		<p>
		First download the R source code from <a href="https://cran.r-project.org/src/base/R-3/R-3.2.2.tar.gz" download>here</a>.
		</p>

		<p>
		Then, move the tgz in the folder where you want to install the library (e.g. $(HOME)/Libraries), and run the following commands:
		</p>

		<code>
		tar xzvf R-3.2.2.tar.gz<br>
		cd R-3.2.2/<br>
		./configure<br>
		cd src/nmath/standalone/<br>
		make
		</code>
		
		<p>
		This generates a library file <b>libRmath.a</b>. If the file is located here: <i>/home/me/Libraries/R-3.2.2/src/nmath/standalone/</i>, then set up <i>RMATH_INC</i> and <i>RMATH_LIB</i> as follows:
		</p>

		<code>
		RMATH_INC=/home/me/Libraries/R-3.2.2/src/include<br>
		RMATH_LIB=/home/me/Libraries/R-3.2.2/src/nmath/standalone
		</code>

		<br>
		<h3>4. The HTSlib library</h3>
		<p> This library written in C allows reading and parsing efficiently VCF, BCF, BAM and BED file. This is used by widely used tools such as samtools or bcftools for example. Download the library from <a href="https://github.com/samtools/htslib/releases/download/1.3.1/htslib-1.3.1.tar.bz2" download>here</a>. Then, move the tgz in the folder you want to install the library and run the following commands:
		</p>
		
		<code>
		tar xjvf htslib-1.3.1.tar.bz2<br>
		cd htslib-1.3.1/<br>
		./configure<br>
		make
		</code>
		
		<p> If you are using HTSlib version 1.4 or above make sure you have the zlib, bzip2, lzma libraries, omit the configuration step (./configure) and type make directly. This will avoid linking in the curl library. Alternatively you can modify the makefile to include this library. </p>  
		
		<p>
		This generates a library file <b>libhts.a</b>. If the file is located here: <i>/home/olivier/Libraries/htslib-1.3.1/</i>, then set up <i>HTSLD_INC</i> and <i>HTSLD_LIB</i> as follows:
		</p>
		
		<code>
		HTSLD_INC=/home/olivier/Libraries/htslib-1.3.1<br>
		HTSLD_LIB=/home/olivier/Libraries/htslib-1.3.1
		</code>

		<br>
		<h3>5. Compile QTLtools</h3>
		<p>
		Once the variables <i>BOOST_INC, BOOST_LIB, RMATH_INC, RMATH_LIB, HTSLD_INC, HTSLD_LIB</i> correctly set up, you should have something like this:
		</p>

		<code>
		########################################################<br>
		# YOU NEED TO UPDATE THESE 6 PATHS TO COMPILE QTLTOOLS #<br>
		########################################################<br>
		BOOST_INC=/usr/include<br>
		BOOST_LIB=/usr/lib/x86_64-linux-gnu<br>
		RMATH_INC=/home/me/Libraries/R-3.2.2/src/include<br>
		RMATH_LIB=/home/me/Libraries/R-3.2.2/src/nmath/standalone<br>
		HTSLD_INC=/home/olivier/Libraries/htslib-1.3.1<br>
		HTSLD_LIB=/home/olivier/Libraries/htslib-1.3.1<br>
		</code>

		<p>
		Then, you can proceed with the compilation using:
		</p>

		<code>
		make
		</code>
	
		<p>This will generate the binary <i>./bin/QTLtools</i>. And you're ready to go!</p>	
		<p>If you have root privileges, you can install QTLtools with:</p>
		<code>
		  make install
		</code>
		<p>This will install the binary, man pages, scripts, and the bash autocomplete script.</p>
		<p>If you do not have root privileges, then you can append <i>./bin/</i> to your $PATH and <i>./man/</i> to your $MANPATH variables. </p>	
	</div>
	<div class="log">Thursday 27th August 2020</div>
</div>
</body>
</html>
