<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
</head>

<div class="sidenav">
	<div class="item">
		<h1>Basic information</h1>
		<ul>
			<li><a href="basic_homepage.html" target='content'>Homepage</a></li>
			<li><a href="basic_install.html" target='content'>Download & Installation</a></li>
			<li><a href="basic_compilation.html" target='content'>Compilation examples</a></li>
			<li><a href="https://github.com/qtltools/qtltools" target='_top'>GitHub / QTLtools</a></li>
			<li><a href="basic_figures.html" target='content'>QTLtools figures</a></li>
		</ul>
	</div>
	
	<div class="item">
		<h1>Analyses</h1>
		<ul>
			<li><a href="input_files.html" target='content'>Preparing input files</a></li>
			<li><a href="mode_overview.html" target='content'>Overview of QTLtools</a></li>
			<li><a href="mode_bamstat.html" target='content'><b>[bamstat]</b> QC the sequence data</a></li>
			<li><a href="mode_match.html" target='content'><b>[mbv]</b> Sequence to genotype matching</a></li>
			<li><a href="mode_quant.html" target='content'><b>[quan]</b> Quantify gene expression</a></li>
			<li><a href="mode_pca.html" target='content'><b>[pca]</b> Perform PCA on data</a></li>
			<li><a href="mode_cis_nominal.html" target='content'><b>[cis]</b> Discover QTL in <i>cis</i> [nominal pass]</a></li>
			<li><a href="mode_cis_permutation.html" target='content'><b>[cis]</b> Discover QTL in <i>cis</i> [permutation pass]</a></li>
			<li><a href="mode_cis_conditional.html" target='content'><b>[cis]</b> Discover QTL in <i>cis</i> [conditional pass]</a></li>
			<li><a href="mode_trans_full.html" target='content'><b>[trans]</b> Discover QTL in <i>trans</i> [full pass]</a></li>
			<li><a href="mode_trans_approx.html" target='content'><b>[trans]</b> Discover QTL in <i>trans</i> [approx. pass]</a></li>
			<li><a href="mode_fdensity.html" target='content'><b>[fdensity]</b> Annotation density at QTLs</a></li>
			<li><a href="mode_fenrich.html" target='content'><b>[fenrich]</b> Enrichment of QTLs in annotations</a></li>
			<li><a href="mode_rtc.html" target='content'><b>[rtc]</b> Overlap QTLs with GWAS hits</a></li>
		</ul>
	</div>
	
	<div class="item">
		<h1>Man Pages</h1>
		<ul>
			<li><a href="QTLtools.1.html" target='content'>Overview of QTLtools</a></li>
			<li><a href="QTLtools-bamstat.1.html" target='content'><b>[bamstat]</b> Calculate basic QC metrics for BAM/SAM</a></li>
			<li><a href="QTLtools-mbv.1.html" target='content'><b>[mbv]</b> Sequence to genotype matching</a></li>
			<li><a href="QTLtools-pca.1.html" target='content'><b>[pca]</b> Perform PCA on data</a></li>
			<li><a href="QTLtools-correct.1.html" target='content'><b>[correct]</b> Covariate correction of BED or VCF file</a></li>
			<li><a href="QTLtools-cis.1.html" target='content'><b>[cis]</b> Discover QTL in <i>cis</i></a></li>
			<li><a href="QTLtools-trans.1.html" target='content'><b>[trans]</b> Discover QTL in <i>trans</i></a></li>
			<li><a href="QTLtools-fenrich.1.html" target='content'><b>[fenrich]</b> Enrichment of QTLs in annotations</a></li>
			<li><a href="QTLtools-fdensity.1.html" target='content'><b>[fdensity]</b> Annotation density at QTLs</a></li>
			<li><a href="QTLtools-rtc.1.html" target='content'><b>[rtc]</b> Colocalization analysis of QTLs and GWAS hits</a></li>
			<li><a href="QTLtools-rtc-union.1.html" target='content'><b>[rtc-union]</b> Find the union of QTLs</a></li>
			<li><a href="QTLtools-extract.1.html" target='content'><b>[extract]</b>Extract all data from the provided files </a></li>
			<li><a href="QTLtools-quan.1.html" target='content'><b>[quan]</b> Quantify gene and exon expression</a></li>
			<li><a href="QTLtools-ase.1.html" target='content'><b>[ase]</b> Measure allele specific expression</a></li>
			<li><a href="QTLtools-rep.1.html" target='content'><b>[rep]</b> Replicate QTLs in an independent dataset</a></li>
			<li><a href="QTLtools-gwas.1.html" target='content'><b>[gwas]</b>Correlate all genotypes with all phenotypes</a></li>
		</ul>
	</div>

	<div class="item">
		<h1>Credits</h1>
		<ul>
			<li><a href="http://funpopgen.unige.ch/members/olivier-delaneau">Dr Olivier DELANEAU, UNIGE</a></li>
			<li><a href="http://funpopgen.unige.ch/members/halit-ongen">Dr Halit ONGEN, UNIGE</a></li>
			<li><a href="http://funpopgen.unige.ch/members/professor-emmanouil-manolis-dermitzakis">Prof. Manolis DERMITZAKIS, UNIGE</a></li>
		</ul>
		<a href="http://funpopgen.unige.ch" target='_top'><img WIDTH=45% src="../img/funpopgen.png"></img></a>
		<a href="http://www.snf.ch/en/Pages/default.aspx" target='_top'><img WIDTH=45% src="../img/snf.jpg"></img></a>
		<a href="http://www.systemsx.ch" target='_top'><img WIDTH=50% src="../img/systemsx.png"></img></a>
		<a href="https://www.nih.gov" target='_top'><img WIDTH=25% src="../img/nih.png"></img></a>
	</div>
	<div class="item">
		<h1>References</h1>
		<ul>
			<li><a href="https://doi.org/10.1038/ncomms15452" target='_top'>QTLtools: a complete tool set for molecular QTL discovery and analysis. <i>Nat Commun</i> <b>8</b>, 15452 (2017).</a></li>
			<li><a href="https://doi.org/10.1038/ng.3981" target='_top'>Estimating the causal tissues for complex traits and diseases. <i>Nat Genet.</i> 2017;<b>49</b>(12):1676-1683.</a></li>			
			<li><a href="https://doi.org/10.1093/bioinformatics/btx074" target='_top'> MBV: a method to solve sample mislabeling and detect technical bias in large combined genotype and sequencing assay datasets. <i>Bioinformatics</i> <b>33</b>(12), 1895 2017. </a></li>
		</ul>

</html>
