<html>
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">

	<div class="item"  id="item1">
		<h1>Sequence data [BAM/SAM]</h1>
		<p>
		Any sequence data needs to be specified in either <b>BAM</b> or <b>SAM</b> format. To validate the file, use <a href="http://www.htslib.org/">samtools</a> as follows:
		</p>

		<code>
		samtools view mySequences.bam | head
		</code>
		
		<p>
		If you can visualize some of your sequences using this command without any errors, this means that QTLtools will also be able to read the file.
		</p>
	</div>

	<div class="item"  id="item1">
		<h1>Genotype data [VCF/BCF]</h1>
		<p>
		Any genotype data needs to be specified using <b>VCF</b> or <b>BCF</b> file format. Hereafter a minimal VCF example:
		</p>
		
		<code>
		##fileformat=VCFv4.1<br>
		#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT  UNR1	UNR2	UNR3	UNR4<br>
		chr7	123	SNP1	A	G	100	PASS	.	GT:DS	0/0:0.001	0/0:0.000	0/1:0.999	1/1:1.999<br>
		chr7	456	SNP2	T	C	100	PASS	.	GT:DS	0/0:0.001	0/0:0.000	0/1:1.100	0/0:0.100<br>
		chr7	789	SNP3	A	T	100	PASS	.	GT:DS	1/1:2.000	0/1:1.001	0/0:0.010	0/1:0.890<br>
		</code>
		
		<p>
		A precise description of this file format can be found <A href=http://vcftools.sourceforge.net/specs.html target='_top'>here</A>.
		</p>

		<p>
		QTLtools needs any VCF/BCF to be indexed. This can be done using <a href="http://www.htslib.org/doc/tabix.html" target='_top'>tabix</a> for VCFs and <a href="https://samtools.github.io/bcftools/bcftools.html target='_top'">bcftools</a> for BCFs.
		</p>

		<p>
		All QTLtools functionalities can use either the <b>GT</b> or <b>DS</b> (i.e. genotype dosage) fields. The <b>DS</b> field can be very usefule to encode imputation uncertainty. Note that the <b>match</b> mode does not work with the <b>DS</b> field and therefore requires the <b>GT</b> to be specified.
		</p>
		
		<p>
		Missing <b>GT</b> entries are encoded with <b>./.</b> and missing <b>DS</b> entries are encoded with <b>.</b>
		</p>

		<p>
		To validate your VCF/BCF file, use <a href="https://samtools.github.io/bcftools/bcftools.html" target='_top'>bcftools</a> as follows:
		</p>

		<code>
		bcftools view myGenotypes.vcf.gz | less -S
		</code>

		<p>
		If you can parse your genotypes using this command without any errors, this means that QTLtools will also be able to read the file.
		</p>		
	</div>

	<div class="item"  id="item2">
		<h1>Phenotype data [BED]</h1>
		<p>
		Phenotype data are specified using an extended UCSC <b>BED</b> format. It is a standard BED file with some additional columns. The missing values should be encoded as NA.
		Hereafter a general example of 4 molecular phenotypes for 4 samples.  
		</p>
		
		<code>
		#Chr	start	end	pid	gid	strand	UNR1	UNR2	UNR3	UNR4
		chr1	99999	100000	pheno1	pheno1	+	-0.50	0.82	-0.71	0.83<br>
		chr1	199999	201000	pheno2	pheno2	+	1.18	-2.84	1.34	-1.56<br>
		chr1	299999	300000	exon1	gene1	+	-1.13	1.18	-0.03	0.11<br>
		chr1	299999	300000	exon2	gene1	+	-1.18	1.32	-0.36	1.26<br>
		</code>
	
		<p>
		This file is <i>TAB</i> delimited. Each line corresponds to a single molecular phenotype.
		The first 6 columns are:
		</p>
		
		<ol>
			<li>Chromosome ID <i>[string]</i></li>
			<li>Start genomic position of the phenotype (here the TSS of gene1) <i>[integer, 0-based]</i></li>
			<li>End genomic position of the phenotype (here the TSS of gene1) <i>[integer, 1-based]</i></li>
			<li>Phenotype ID (here the exon IDs) <i>[string]</i>.</li>
			<li>Phenotype group ID (here the gene IDs, multiple exons belong to the same gene) <i>[string]</i></li>
			<li>Strand orientation <i>[+/-]</i></li>
		</ol>
		
		<p>
		Then each additional column gives the quantification for a sample. Quantifications are encoded with floating numbers. 
		This file should have P lines and N+6 columns where P and N are the numbers of phenotypes and samples, respectively.
		</p>

		<note>
		THIS FILE FORMAT EXTENDS THE FASTQTL FILE FORMAT BY ADDING 2 COLUMNS! THIS MEANS THAT QTLTOOLS CANNOT WORK WITH FASTQTL BED FILES!
		</note>
		
		<p>
			To make a quick and dirty conversion, you can use this command:
		</p>

		<code>
		zcat myFastQTLphenotypes.bed.gz | awk '{ $4=$4" . +"; print $0 }' | tr " " "\t" | bgzip -c > myQTLtoolsPhenotypes.bed.gz
		</code>

		<p>
		The small example above gives 3 different ways of encoding phenotype data:
		</p>

		<ol>
			<li>pheno1/pheno1 [line1]: the most standard way for encoding a molecular phenotype. It has a unique ID (pheno1) and spans 1bp.</li>
			<li>pheno2/pheno2 [line2]: alternative way of specifying a molecular phenotype. It has a unique ID (pheno2) but spans a region of 1kb.</li>
			<li>gene1 [line3-4]: these two lines specify a group (gene1) of 2 molecular phenotypes (exon1 and exon2). Importantly, both phenotypes need to share the same coordinate otherwise QTLtools will not be able to determine that they belong to the same group.</li>
		</ol>
 
		<p>
		Sample IDs are specified in the header line. This line needs to start with a hash key (i.e. #).
		</p>

		<p>
		This <b>BED</b> file needs to be indexed with <a href="http://www.htslib.org/doc/tabix.html" target='_top'>tabix</a> as follows:
		</p>
		
		<code>
		bgzip myPhenotypes.bed && tabix -p bed phenotypes.bed.gz
		</code>
		
		<p>
		If this doesn't work, this probably means that your BED file is not sorted, so sort it using <i>sort</i>.
		</p>
	</div>
	
	<div class="item"  id="item2">
		<h1>Important notes (to be read!)</h1>
		<p>
		MAKE SURE THAT THE CHROMOSOME IDS ARE THE SAME ACROSS ALL FILES. A very common mistake is to have chromsomes as 1-22 in the genotype file while they are chr1-chr22 in the phenotype or sequence data. In such cases, QTLtools will not be able to find correspondance.
		</p>
	</div>
	
	<div class="item"  id="item2">
		<h1>Covariate data [TXT]</h1>
		<p>
		The <b>COV</b> file contains the covariate data in simple <b>TXT</b> format. The missing values should be encoded as NA. 
		Hereafter an example of 4 covariates for 4 samples.  
		</p>
		
		<code>
		id	UNR1	UNR2	UNR3	UNR4<br>
		PC1	-0.02	0.14	0.16	-0.02<br>
		PC2	0.01	0.11	0.10	0.01<br>
		PC3	0.03	0.05	0.08	0.07<br>
		BIN	1	0	0	1
		</code>
		
		<p>
		Herafter, some properties of this file:
		<ol>
			<li>The file is <i>TAB</i> delimited</li>
			<li>First row gives the sample ID and each additional one corresponds to a single covariate</li> 
			<li>First column gives the covariate ID and each additional one corresponds to a sample</li>
			<li>The file should have S+1 rows and C+1 columns where S and C are the numbers of samples and covariates, respectively.</li>
		</ol>
		
		<p>
		Both quantitative and qualitative covariates are supported.
		<b>Quantitative</b> covariates are assumed when only <b>numeric</b> values are provided.
		<b>Qualitative</b> covariates are assumed when only <b>non-numeric</b> values are provided.
		In practice, qualitative covariates with <i>F</i> factors are converted in <i>F-1</i> binary covariates.
		</p>
	</div>
	
		<div class="item"  id="item2">
		<h1>Include/Exclude file formats</h1>
		<p>
			The various --{include,exclude}-{sites,samples,phenotypes,covatiates} options require a simple text file which lists the IDs of the desired type, one ID per line. The include options will result in running the analyses only in this subset of IDs, whereas exclude options will remove these IDs from the analyses. The IDs for --{include,exclude}-sites refer to the 3rd column in VCF/BCF files, --{include,exclude}-covariates refer to the 1st column in COV files, --{include,exclude}-phenotyps refer to the 4th column in BED files and when --grp-best option is used to the 5th column.  
		</p>
		<p> The --include-positions and --exclude-positions options require a text file which lists the chromosomes and positions (separated by a space) of genotypes to be excluded or included. One position per line.</p>

	</div>
	
	<div class="log">Monday 2nd September 2019</div>
</div>
</body>
</html>

