<html>
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="item1">
		<h1>How to run QTLtools</h1>
		<p>
		QTLtools contain the following multiple functionalities:
		</p>
		
		<ul>
			<li><a href="mode_bamstat.html" target='content'><b>[bamstat]</b></a> to control the quality of the sequence data</li>
			<li><a href="mode_match.html" target='content'><b>[match]</b></a> to ensure good matching between sequence and genotype data</li>
			<li><a href="mode_quant.html" target='content'><b>[quan]</b></a> to quantify gene expression</li>
			<li><a href="mode_pca.html" target='content'><b>[pca]</b></a> to perform PCA on phenotype and genotype data</li>
			<li><a href="mode_cis_nominal.html" target='content'><b>[cis]</b></a> to discover QTL in <i>cis</i> via a nominal pass of association</li>
			<li><a href="mode_cis_permutation.html" target='content'><b>[cis]</b></a> to discover QTL in <i>cis</i> via a permutation pass of association</li>
			<li><a href="mode_cis_conditional.html" target='content'><b>[cis]</b></a> to discover multiple QTLs per phenotype in <i>cis</i> using a conditional pass</li>
			<li><a href="mode_trans_full.html" target='content'><b>[trans]</b></a> to discover QTL in <i>trans</i> using a full permutation pass</li>
			<li><a href="mode_trans_approx.html" target='content'><b>[trans]</b></a> to discover QTL in <i>trans</i> using an approximated permutation pass</li>
			<li><a href="mode_fdensity.html" target='content'><b>[fdensity]</b></a> to measure functional annotation density around QTLs</li>
			<li><a href="mode_fenrich.html" target='content'><b>[fenrich]</b></a> to measure enrichment of QTLs within annotations</li>
			<li><a href="mode_rtc.html" target='content'><b>[rtc]</b></a> to overlap QTLs with GWAS hits</li>
		</ul>

		<p>
		To get the full list of available nodes, use:
		</p>

		<code>
		QTLtools
		</code>

		<note>
		QTLtools is under constant development and multiple modes are experimental. We do not recommend to use a mode that is not described in this online documentation
		</note>

		<p>
		To run a particular functionality, you need to combine QTLtools with the appropriate mode. For instance, to run <i>bamstat</i>, use:
		</p>

		<code>
		QTLtools bamstat ...
		</code>

		<p>
		Where ... stands for all options unique to the mode used. To get a full list of available options with short description, run:
		</p>

		<code>
		QTLtools bamstat --help
		</code>

		<p>
		QTLtools allows dumping the screen output in a <i>LOG</i> file. To do so, use:
		</p>

		<code>
		QTLtools mode ... --log myScreenOutput.log
		</code>

		<p>
		For any procedure based on random permutations, you can specify the seed in order to get a different random number sequence or to reproduce a particular experiment. To do so, use:
		</p>

		<code>
		QTLtools mode ... --seed 12345
		</code>
	</div>
	
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
