<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="item1">
		<h1>How to run pca?</h1>
		<p>
		The QTLtools mode pca allows performing a <a href="https://en.wikipedia.org/wiki/Principal_component_analysis" target='_top'>Principal Component Analysis (PCA)</a> either on molecular phenotype quantifications or genotype data. It is typically used (i) to detect outliers in the data, (ii) to detect stratification in the data or (iii) to build a covariate matrix before QTL mapping. 
		</p>

		<br>
		<h3>Phenotype data</h3>
		<p>
		In this case, it requires just a single file as input:
		</p>

		<ul>
			<li>A <i>BED</i> file containing the quantification data</li>
		</ul>
		
		<p>
		To illustrate how this works, we provide the following example file:
		</p>

		<ul>
			<li>A <i>BED</i> file of gene-level RPKM values for 358 samples on chr22: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
		</ul>
		
		<p>
		Then, to perform PCA on this file, use:
		</p>

		<code>
		QTLtools <b>pca</b> --bed genes.50percent.chr22.bed.gz --scale --center --out genes.50percent.chr22
		</code>
		
		<p>
		This is going to produce two files:
		</p>

		<ul>
			<li><i>genes.50percent.chr22.pca</i> that contains the individual coordinates on the Principal Components (PCs)</li>
			<li><i>genes.50percent.chr22.pca_stats</i> that contains the percentages of the variance explained by each PC</li>
		</ul>

		<p>
		If you open <i>genes.50percent.chr22.pca</i>, you'll get something that looks like this:
		</p>

		<code>
		SampleID	HG00096	HG00097	HG00099	HG00100<br>
		genes.50percent.chr22_1_1_svd_PC1	9.4607	3.61907	-7.27784	-4.74021<br>
		genes.50percent.chr22_1_1_svd_PC2	-2.89575	-6.1823	-0.488397	-3.1485<br>
		genes.50percent.chr22_1_1_svd_PC3	-3.24429	-4.06885	-6.76097	-0.231029<br>
		genes.50percent.chr22_1_1_svd_PC4	-0.00959694	2.1984	1.58854	-3.01095<br>
		genes.50percent.chr22_1_1_svd_PC5	1.50254	-1.01627	2.46998	4.49168<br>
		genes.50percent.chr22_1_1_svd_PC6	-1.06957	-0.953119	3.60855	2.53457<br>
		genes.50percent.chr22_1_1_svd_PC7	-6.21865	2.10234	-5.26164	3.38665<br>
		genes.50percent.chr22_1_1_svd_PC8	2.62994	-0.516754	0.234968	1.15582<br>
		genes.50percent.chr22_1_1_svd_PC9	-3.62158	0.246679	-8.15425	-1.29294<br>	
		...
		</code>

		<p>
		In this file, the header line gives the sample IDs, the first column the ID of the PCs, starting from the first one and each successive line the coordinates of the samples on the PCs. The second output file <i>genes.50percent.chr22.pca_stats</i> looks like this:
		</p>

		<code>
		sd	7.81917	7.40651	6.66507	5.40993	4.96369	4.56251	4.41553	3.62532	3.34306<br>
		prop_var	0.100544	0.0902117	0.0730544	0.0481304	0.0405178	0.0342329	0.0320629	0.0216137	0.0183791<br>
		cumm_prop	0.100544	0.190756	0.26381	0.311941	0.352459	0.386692	0.418755	0.440368	0.458747<br>
		</code>

		<p>The 3 first lines give you for each successive PC:</p>

		<ul>
			<li>1. The standard deviation</li>
			<li>2. The variance explained</li>
			<li>3. The cumulative variance explained</li>
		</ul>

		<p>
		The options <i>--center</i> and <i>--scale</i> can be used to enforce centering and scaling of the phenotype values prior to the PCA.
		</p>

		<br>
		<h3>Genotype data</h3>
		<p>
		To illustrate how this works, we provide the following example file:
		</p>

		<ul>
			<li>A <i>VCF</i> file of genotypes for 358 samples on chr22: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>VCF</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.tbi" download>index</a></li>
		</ul>

		<p>
		Then, to perform PCA on this file, use:
		</p>

		<code>
		QTLtools pca <b>--vcf genotypes.chr22.vcf.gz</b> --scale --center <b>--maf 0.05 --distance 50000</b> --out genotypes.chr22
		</code>

		<p>
		The output files are the same than for phenotype data. The only difference here is that we added the two following options to ensure that we have variant sites in linkage equilibrium:
		</p>

		<ul>
			<li><i>--maf 0.05</i> to only consider variant sites with a Minor Allele Frequency (MAF) above 5%</li>
			<li><i>--distance 50000</i> to only consider variant sites separated by at least 50kb</li>
		<ul>
	</div>

	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
