<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="cis_map1">
		<h1>How to run a nominal pass in cis?</h1>
		<p>
		To perform a simple nominal pass on your data set in order to get nominal P-values between your phenotypes and genotypes, you need the following input files:
		</p>

		<ul>
			<li>An indexed phenotype data matrix in <i>BED</i> format (see the section: Preparing input files)</li>
			<li>An indexed genotype data matrix in <i>VCF/BCF</i> format (see the section: Preparing input files)</li>
			<li>A covariate data matrix in <i>TXT</i> format (see the section: Preparing input files)</li>
		</ul>

		<p>
		For simplicitly, we provide the following small example files:
		</p>

		<ul>
			<li> The phenotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
			<li> The genotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>VCF</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.tbi" download>index</a></li>
			<li> The covariate data matrix on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.covariates.pc50.txt.gz" download>TXT</a></li>
		</ul>

		<p>
		Then, you can run a nominal pass of the data in a particular genomic region using:
		</p>

		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--nominal 0.01</b> --region chr22:17000000-18000000 --out nominals.txt
		</code>

		<p>
		The given options are:
		</p>

		<ul>
			<li><i>--nominal 0.01</i> to only output phenotype-variant pairs with a nominal p-value below 0.01. If you want the full output, use instead <i>--nominal 1</i> which is going to create big files </li>
			<li><i>--region chr22:17000000-18000000</i> to only process phenotypes which locations are within the specified genomic region</li>
		</ul>

		<p>The output file <i>nominals.txt</i> looks like this:</p>

		<code>
		ENSG00000237438.2 chr22 17517460 17517460 + 5803 -202619 rs74762450 chr22 17314841 17314841 0.00957345 -0.347492 0<br>
		ENSG00000237438.2 chr22 17517460 17517460 + 5803 -165953 rs5748687 chr22 17351507 17351507 0.00227542 -0.225077 0<br>
		ENSG00000237438.2 chr22 17517460 17517460 + 5803 -165363 rs4006343 chr22 17352097 17352097 0.0022876 -0.226253 0<br>
		ENSG00000237438.2 chr22 17517460 17517460 + 5803 -164231 rs3875996 chr22 17353229 17353229 0.00181073 -0.233649 0<br>
		...<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 469530 rs5992121 chr22 18310367 18310367 0.00653855 -0.00519037 0<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 473554 rs367922 chr22 18314391 18314391 0.00125305 -0.00612444 0<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 476586 rs11705197 chr22 18317423 18317423 0.00961104 -0.00498535 0<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 754091 rs362027 chr22 18594928 18594928 0.00673601 0.0100518 0<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 754094 rs361893 chr22 18594931 18594931 0.00673601 0.0100518 0<br>
		</code>

		<p>
		The columns are:
		</p>

		<ul>
			<li>1. The phenotype ID</li>
			<li>2. The chromosome ID of the phenotype</li>
			<li>3. The start position of the phenotype</li>
			<li>4. The end position of the phenotype</li>
			<li>5. The strand orientation of the phenotype</li>
			<li>6. The total number of variants tested in <i>cis</i></li>
			<li>7. The distance between the phenotype and the tested variant (accounting for strand orientation)</li>
			<li>8. The ID of the tested variant</li>
			<li>9. The chromosome ID of the variant</li>
			<li>10. The start position of the variant</li>
			<li>11. The end position of the variant</li>
			<li>12. The nominal P-value of association between the variant and the phenotype</li>
			<li>13. The corresponding regression slope</li>
			<li>14. A binary flag equal to 1 is the variant is the top variant in <i>cis</i></li>
		</ul>

		<p>
		Associations between genotype dosages and phenotype quantifications are measured with linear regressions, similar to the R/lm function. This model assumes that phenotypes are normally distributed. To enforce them to match normal distributions N(0, 1), we recommend to use:
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --nominal 0.01 --region chr22:17000000-18000000 --out nominals.txt <b>--normal</b>
		</code>
		
		<p>
		To change the size of the cis-window (i.e. the maximal distance spanned by phenotype-variant pairs to be considered for testing) from default value 1e6 bp to 2e6 bp, use: 
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --nominal 0.01 --region chr22:17000000-18000000 --out nominals.txt <b>--window 2000000</b>
		</code>

		<p>
		To exclude samples, variants, phenotypes or covariates from the analysis, you can use one of these options:
		</p>
		
		<ol>
			<li>To exclude samples: <i>--exclude-samples file.exc</i></li>
			<li>To exclude variants: <i>--exclude-sites file.exc</i></li>
			<li>To exclude phenotypes: <i>--exclude-phenotypes file.exc</i></li>
			<li>To exclude caovariates: <i>--exclude-covariates file.exc</i></li>
		</ol>
		
		<p>
		For instance, if you want to ignore these 3 samples when analyzing the example data set (HG00277, HG00278, HG00280), create a text file containing the IDs of the samples to be excluded, called here <i>file.exc</i>:   
		</p>
		
		<code>
		HG00277<br>
		HG00278<br>
		HG00280<br>
		</code>
		
		<p> 
		Then, run:
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --nominal 0.01 --region chr22:17000000-18000000 --out nominals.txt <b>--exclude-sample file.exc</b>
		</code>
		
		<p>
		Similarly to these 4 options for data exclusion, you can also specify the set of samples, variants, phenotypes and covariates you wish to include in the analysis using the options: <i>--include-samples, --include-sites, --include-phenotypes and --include-covariates</i>, respectively.
		</p>

		<br>
		<h3>Important notes on parallelization</h3>
		<p>
		To facilitate parallelization on compute cluster, we developed an option to run the analysis into chunks of molecular phenotypes.
		This is quite similar to what is commonly used for genotype imputation, where only small chunks of data are imputed one at a time in seperate jobs.
		However in practice, it is usually quite complicated to properly split the genome into a given number of chunks with correct coordinates.
		To facilitate this, we embedded all coordinates into a chunk-based system so that you only need to specify the chunk index you want to run. 
		Then, splitting the genome into chunks, extraction of data, and analysis are automatically performed.
		For instance, to run analysis on chunk number 12 when splitting the example data set into 20 chunks, just run:   		   
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --nominal 0.01 <b>--chunk 12 20</b> --out nominals_12_20.txt
		</code>
		
		<p>
		If you want to submit the whole analysis into 20 jobs on a compute cluster, just run:
		</p>
		
		<code>
		for j in <b>$(seq 1 20)</b>; do<br>
		&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;echo "QTLtools cis --vcf ... --bed ... --cov ... --nominal 0.01 <b>--chunk $j 20</b> --out nominals_$j\_20.txt" | qsub<br>
		done
		</code>
		
		<p>
		Here <b>qsub</b> needs to be changed according to the job submission system used (bsub, psub, etc...).  
		</p>
		
		<note>
		In this simple example, we only split the data into 20 chunks. However, a realistic whole genome analysis would require to split the data in 200, 500 or even 1,000 chunks to fully use the capabilities of a modern compute cluster.  
		</note>
	</div>
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
