<html>
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>FastQTL</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="cis_perm1">
		<h1>How to run a conditional analysis in cis?</h1>
		<p>
		A conditional analysis can only be run once a permutation pass on the data has been done. Why? Because the permutation pass allows to get a nominal P-value threshold of significance per phenotypes.
		To illustrate this, first download the example data below and go through the two steps; 1. permutation pass and 2. conditional analysis.
		</p>

		<ul>
			<li> The phenotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
			<li> The genotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>VCF</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.tbi" download>index</a></li>
			<li> The covariate data matrix on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.covariates.pc50.txt.gz" download>TXT</a></li>
		</ul>

		<br>
		<h3>Step1: Run the permutation pass</h3>

		<p>
		This can be done quickly using 4 threads as follows:
		</p>

		<code>
		for j in $(seq 1 16); do<br>
		&nbsp;&nbsp;echo "cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --permute 200 --chunk $j 16 --out permutations_$j\_16.txt";<br>
		done | xargs -P4 -n14 QTLtools<br>
		</code>

		<p>
		Then, cat all chunks together and run FDR correction:
		</p>

		<code>
		cat permutations_*.txt | gzip -c > permutations_all.txt.gz<br>
		Rscript ./script/runFDR_cis.R permutations_all.txt.gz 0.05 permutations_all
		</code>

		<p>
		From this, you get a file <i>permutations_all.thresholds.txt</i> with the nominal P-value thresholds that you need for the conditional analysis:
		</p>

		<code>
		ENSG00000182057.4 7.6768047839515e-05<br>
		ENSG00000233903.2 9.8313776034753e-05<br>
		ENSG00000273366.1 8.39380360819764e-05<br>
		...<br>
		ENSG00000225783.2 2.72012352409146e-05<br>
		ENSG00000206028.1 3.83673656872974e-05<br>
		ENSG00000244625.1 3.66061481116287e-05
		</code>
	
		<br>
		<h3>Step2: Run the conditional analysis</h3>
		<p>
		Now you can proceed with the actual conditional analysis. For one chunk, this is done as:
		</p>

		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--mapping permutations_all.thresholds.txt</b> --chunk 12 16 --out conditional_12_16.txt
		</code>

		<p>
		The output <i>conditional_12_16.txt</i> looks like this: 
		</p>

		<code>
		ENSG00000073150.9 chr22 50609160 50609160 + 5641 -71259 rs12169319 chr22 50537901 50537901 0 6.60258e-08 0.202819 1 1 6.60258e-08 0.202819 1 1<br>
		ENSG00000073150.9 chr22 50609160 50609160 + 5641 -58747 rs9617077 chr22 50550413 50550413 0 6.84017e-05 0.0802672 0 1 6.84017e-05 0.0802672 0 1<br>
		ENSG00000073150.9 chr22 50609160 50609160 + 5641 -53474 rs760749 chr22 50555686 50555686 0 3.83196e-05 0.083734 0 1 3.83196e-05 0.083734 0 1<br>
		...<br>
		ENSG00000184319.11 chr22 51222185 51222185 + 3201 11115 rs62240042 chr22 51233300 51233300 0 9.89086e-06 -0.105956 0 1 0.000483508 -0.292154 0 0<br>
		ENSG00000184319.11 chr22 51222185 51222185 + 3201 -41251 rs9616824 chr22 51180934 51180934 1 4.9067e-06 -0.390492 1 1 4.9067e-06 -0.390492 1 1<br>
		ENSG00000184319.11 chr22 51222185 51222185 + 3201 -29599 rs5771006 chr22 51192586 51192586 1 7.71478e-05 -0.391773 0 1 7.71478e-05 -0.391773 0 1<br>
		</code>

		<p>
		With the various columns giving:
		</p>

		<ul>
			<li>1. Phenotype ID</li>
			<li>2. Phenotype chr ID</li>
			<li>3. Phenotype start position</li>
			<li>4. Phenotype end position</li>
			<li>5. Phenotype strand orientation</li>
			<li>6. Number of variants tested in <i>cis</i></li>
			<li>7. Distance between the variant and the phenotype</li>
			<li>8. Variant ID</li>
			<li>9. Variant chr ID</li>
			<li>10. Variant start position</li>
			<li>11. Variant end position</li>
			<li>12. <b>Rank of the association</b>. This tells you if the variant has been mapped as belonging to the best signal (rank=0), the second best (rank=1), etc ... As a consequence, the maximal rank value for a given phenotype tells you how many independent signals there are (e.g. rank=2 means 3 independent signals).</li>
			<li>13. Forward nominal P-value</li>
			<li>14. Forward regression slope</li>
			<li>15. Binary flag; 1 means that the variant is the top forward variant of this rank, 0 otherwise</li>
			<li>16. Binary flag; 1 means that the forward P-value is below the threshold of this phenotype, 0 otherwise</li>
			<li>17. Backward nominal P-value</li>
			<li>18. Backward regression slope</li>
			<li>19. Binary flag; 1 means that the variant is the top backward variant of this rank, 0 otherwise</li>
			<li>20. Binary flag; 1 means that the backward P-value is below the threshold of this phenotype, 0 otherwise</li>
		</ul>

		<note>
		We strongly recommend to use the outcome of the backward pass for any downstrean analysis, since it represents what is given by a full linear model incorporating all signals together.
		The forward is just there for information.
		</note>

		<p>
		Now, run all chunks and if you are only interested in the top variant for each signal, you can filter out the results for the column 19 as follows:
		</p>

		<code>
		cat conditional_full.txt | awk '{ if ($19 == 1) print $0}' > conditional_top_variants.txt
		</code>
	
		<p>
		Let's take an example:
		</p>
		
		<code>
		cat conditional_top_variants | grep ENSG00000188130.9<br>
		ENSG00000188130.9 chr22 50700254 50700254 - 5139 -93 rs11913279 chr22 50700347 50700347 <b>0</b> 2.25986e-09 -0.278047 1 1 1.58737e-12 -0.724656 1 1<br>
		ENSG00000188130.9 chr22 50700254 50700254 - 5139 94985 rs116914202 chr22 50605269 50605269 <b>1</b> 7.6825e-05 -0.533554 0 1 7.58821e-06 -0.602191 1 1<br>
		ENSG00000188130.9 chr22 50700254 50700254 - 5139 65061 rs4084288 chr22 50635193 50635193 <b>2</b> 7.29639e-05 -0.347478 1 1 7.29639e-05 -0.347478 1 1<br>
		</code>

		<p>
		From this, you can see that the gene <i>ENSG00000188130.9</i> has three significant eQTLs with independent effects: <i>rs11913279</i> (rank=0), <i>rs116914202</i> (rank=1) and <i>rs4084288</i> (rank=2).
		</p>
	</div>
	
	<div class="log"><script type="text/javascript"> print_last_modif_date("$Date: 2015-04-01 16:11:24 +0200 (Wed, 01 Apr 2015) $"); </script></div>
</div>
</body>
</html>
