<html>
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>FastQTL</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="cis_perm1">
		<h1>How to run a permutation pass in cis?</h1>
		<p>
		To perform a permutation based analysis on your data set in order to get adjusted P-values of association between the phenotypes and the top variants in <i>cis</i>, you need the following input files:
		</p>

		<ul>
			<li>An indexed phenotype data matrix in <i>BED</i> format (see section <a href="input_files.html" target='content'>Preparing input files</a>)</li>
			<li>An indexed genotype data matrix in <i>VCF/BCF</i> format (see section <a href="input_files.html" target='content'>Preparing input files</a>)</li>
			<li>A covariate data matrix in <i>TXT</i> format (see section <a href="input_files.html" target='content'>Preparing input files</a>)</li>
		</ul>

		<br>
		<h3>1. Standard permutation pass</h3>

		<p>
		For simplicitly, we provide the following small example files:
		</p>

		<ul>
			<li> The phenotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
			<li> The genotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>VCF</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.tbi" download>index</a></li>
			<li> The covariate data matrix on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.covariates.pc50.txt.gz" download>TXT</a></li>
		</ul>

		<p>
		Then, you can run a permutation pass of the data in a particular genomic region using:
		</p>

		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--permute 1000</b> --region chr22:17000000-18000000 --out permutations.txt
		</code>

		<p>
		The given options are:
		</p>

		<ul>
			<li><i>--permute 1000</i> to perform 1,000 permutations of the phenotypes corrected for covariates </li>
			<li><i>--region chr22:17000000-18000000</i> to only process phenotypes which locations are within the specified genomic region</li>
		</ul>

		<p>The output file <i>permutations.txt</i> looks like this:</p>

		<code>
		ENSG00000237438.2 chr22 17517460 17517460 + 5803 25350 rs2845402 chr22 17542810 17542810 356 307.641 1.0758 606.072 1.1864e-25 -0.751804 0.000999001 4.79271e-21<br>
		ENSG00000273203.1 chr22 17548711 17548711 + 5889 -31323 rs3016112 chr22 17517388 17517388 356 305.346 1.05214 604.311 5.86611e-06 -0.140609 0.00999001 0.0128458<br>
		ENSG00000273442.1 chr22 17561591 17561591 + 5911 -8065 rs73149812 chr22 17553526 17553526 356 308.754 1.08518 661.618 0.000244522 0.134045 0.318681 0.304744<br>
		ENSG00000177663.9 chr22 17565844 17565844 + 5919 -52141 rs2908526 chr22 17513703 17513703 356 333.628 0.998034 982.885 8.03358e-14 -0.209617 0.000999001 4.92276e-10<br>
		ENSG00000183307.3 chr22 17602257 17602257 - 5967 14282 rs6518661 chr22 17587975 17587975 356 335.297 0.983505 921.283 1.56789e-07 -0.00776053 0.000999001 0.00038022<br>
		ENSG00000069998.8 chr22 17646177 17646177 - 6044 3400 rs71200232 chr22 17642776 17642777 356 332.081 1.03052 958.093 8.87771e-17 -1.49798 0.000999001 3.8205e-13<br>
		ENSG00000093072.11 chr22 17702879 17702879 - 6048 3249 rs5747027 chr22 17699630 17699630 356 287.216 1.02518 511.292 0.000578649 0.640374 0.628372 0.629492<br>
		ENSG00000099954.14 chr22 17840837 17840837 + 6200 -664900 rs138493943 chr22 17175937 17175937 356 237.501 0.927455 289.487 0.000866455 0.00673291 0.874126 0.86718
		</code>

		<p>
		The columns are:
		</p>

		<ul>
			<li>1. The phenotype ID</li>
			<li>2. The chromosome ID of the phenotype</li>
			<li>3. The start position of the phenotype</li>
			<li>4. The end position of the phenotype</li>
			<li>5. The strand orientation of the phenotype</li>
			<li>6. The total number of variants tested in <i>cis</i></li>
			<li>7. The distance between the phenotype and the tested variant (accounting for strand orientation)</li>
			<li>8. The ID of the top variant</li>
			<li>9. The chromosome ID of the top variant</li>
			<li>10. The start position of the top variant</li>
			<li>11. The end position of the top variant</li>
			<li>12. The number of degrees of freedom used to compute the P-values</li>
			<li>13. Dummy</li>
			<li>14. The first parameter value of the fitted beta distribution</li>
			<li>15. The second parameter value of the fitted beta distribution (it also gives the effective number of independent tests in the region)</li>
			<li>16. The nominal P-value of association between the phenotype and the top variant in <i>cis</i></li>
			<li>17. The corresponding regression slope</li>
			<li>18. The P-value of association adjusted for the number of variants tested in <i>cis</i> given by the direct method (i.e. empirircal P-value)</li>
			<li>19. The P-value of association adjusted for the number of variants tested in <i>cis</i> given by the fitted beta distribution. We strongly recommend to use this adjusted P-value in any downstream analysis</li>
		</ul>
		
		<p>
		If needed, you can either increase or decrease the number of permutations using (here set to 10,000): 
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--permute 10000</b> --region chr22:17000000-18000000 --out permutations.txt
		</code>

		<p>
		In order to perform a fast permutation pass for testing purpose, you can use <i>--permute 100</i> which will already give reasonnable estimate of adjusted P-values via beta distributions.
		</p>
		
		<p>
		Similarly to a nominal pass, you can enforce your phenotypes to match normal distributions N(0, 1) and change the cis-window size by using:
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--permute 1000</b> --region chr22:17000000-18000000 --out permutations.txt <b>--normal --window 2000000</b>
		</code>
		
		<p>
		To change the seed of the random number generator, which is particularly useful to replicate analyses, use:
		</p>
		
		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz <b>--permute 1000</b> --region chr22:17000000-18000000 --out permutations.txt <b>--seed 123456</b>
		</code>

		<p>
		Here also, samples, variants, phenotypes or covariates can be excluded from the analysis using one of these options (see the section about the nominal pass for more details):
		</p>

		<ol>
			<li>To exclude samples: <i>--exclude-samples file.exc</i></li>
			<li>To exclude variants: <i>--exclude-sites file.exc</i></li>
			<li>To exclude phenotypes: <i>--exclude-phenotypes file.exc</i></li>
			<li>To exclude caovariates: <i>--exclude-covariates file.exc</i></li>
		</ol>

		<br>
		<h3>2. Important notes on parallelization</h3>
		<p>
		To facilitate parallelization on compute cluster, we developed an option to run the analysis into chunks of molecular phenotypes.
		This is quite similar to what is commonly used for genotype imputation, where only small chunks of data are imputed one at a time in seperate jobs.
		However in practice, it is usually quite complicated to properly split the genome into a given number of chunks with correct coordinates.
		To facilitate this, we embedded all coordinates into a chunk-based system so that you only need to specify the chunk index you want to run. 
		Then, splitting the genome into chunks, extraction of data, and analysis are automatically performed.
		For instance, to run analysis on chunk number 12 when splitting the example data set into 20 chunks, just run:   		   
		</p>
		
		<code>http://jungle.unige.ch/QTLtools_examples/exons.50percent.chr22.bed.gz
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --permute 1000 <b>--chunk 12 20</b> --out permutations_12_20.txt
		</code>
		
		<p>
		If you want to submit the whole analysis into 20 jobs on a compute cluster, just run:
		</p>
		
		<code>
		for j in <b>$(seq 1 20)</b>; do<br>
		&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;echo "QTLtools cis --vcf ... --bed ... --cov ... --permute 1000 <b>--chunk $j 20</b> --out permutations_$j\_20.txt" | qsub<br>
		done
		</code>
		
		<p>
		Here <b>qsub</b> needs to be changed according to the job submission system used (bsub, psub, etc...).  
		</p>
		
		<note>
		In this simple example, we only split the data into 20 chunks. However, a realistic whole genome analysis would require to split the data in 200, 500 or even 1,000 chunks to fully use the capabilities of a modern compute cluster.  
		</note>

		<p>
		Once all you chunks are processed, you can then merge all of them to build a unique output file using:
		</p>

		<code>
		cat permutations_*_20.txt | gzip -c > permutations_full.txt.gz
		</code>


		<br>
		<h3>3. Check that the beta approximated P-values are okay</h3>
		
		<p>
		To check that the beta approximated permutation p-values are well calibrated, load the data in R and make the following plot:
		</p>
		
		<code>
		R> d = read.table("permutations_full.txt.gz", hea=F, stringsAsFactors=F)<br>
		R> plot(d$V18, d$V19, xlab="Direct method", ylab="Beta approximation", main="")<br>
		R> abline(0, 1, col="red")<br>
		</code>

		<p>
		You should get something like this:
		</p>
				
		<img WIDTH=20% src="../img/fastqtl_checks.jpg"></img>
				
		<br>
		<p>
		We expect here to get all points along the diagonal as on the plot above. This shows that unsignificant beta approximated p-values are well calibrated given the empirical ones.
		</p>

		<br>
		<h3>4. Correct for the multiple phenotypes being tested</h3>
		<p>
		To account for the fact that multiple molecular phenotypes are tested whole genome, we need an additional level of multiple testing correction. 
		To do so, we recommend to use the Storey & Tibshirani <a href="http://en.wikipedia.org/wiki/False_discovery_rate">False Discovery Rate</a> procedure implemented in the <a href="http://svitsrv25.epfl.ch/R-doc/library/qvalue/html/qvalue.html">R/qvalue package</a> as shown below:
		</p>

		<p>
		First, download this example file that contains results for a whole genome analysis for 358 samples and 22,147 genes: <a href="http://jungle.unige.ch/QTLtools_examples/results.genes.full.txt.gz">DATA</a>. Then, use R as follows:
		</p>

		<code>
		R> library(qvalue)
		R> d = read.table("results.genes.full.txt.gz", hea=FALSE, stringsAsFactors=FALSE)<br>
		R> d=d[!is.na(d$V19),]<br>
		R> d$qval=qvalue(d$V19)$qvalue<br>
		R> write.table(d[which(d$qval <= 0.05), ], "results.genes.significant.txt", quote=FALSE, row.names=FALSE, col.names=FALSE)
		</code>

		<p>
		This produces an output file <i>results.genes.significant.txt</i> that contains all significant phenotypes at 5% FDR in <i>results.genes.full.txt.gz</i>.
		For convenience, we also provide a R script <i>runFDR.R</i> located in the <i>script</i> folder of QTLtools that does this trick. To produce the output file <i>results.genes.significant.txt</i> at 5% FDR, use it as follows:
		</p>

		<code>
		Rscript ./script/runFDR_cis.R results.genes.full.txt.gz <b>0.05</b> results.genes
		</code>

		<p>
		Note that alternatively to R/qvalue, you can also use the various FDR procedures implemented in R/p.adjust.
		</p>

		<br>
		<h3>5. Map QTLs for groups of phenotypes</h3>
		<p>
		When your phenotypes in the BED file are grouped (see section <a href="input_files.html" target='content'>Preparing input files</a>), you can perform a permutation pass at the phenotype group level in order to discover group-level QTLs.
		To illustrate this, please download a phenotype matrix that contains groups (i.e. genes) of phenotypes (i.e. exons): <a href="http://jungle.unige.ch/QTLtools_examples/exons.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/exons.50percent.chr22.bed.gz.tbi" download>index</a>. Then, you can run a permutation for all genes in the genomic region [chr22:17000000-18000000] using:
		</p>

		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed exons.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --permute 1000 --region chr22:17000000-18000000 <b>--grp-best</b> --out permutations.group.txt
		</code>

		<p>
		This will produce the following output file <i>permutations.group.txt</i>:
		</p>

		<code>
		ENSG00000237438.2 chr22 17517460 17517460 + ENSG00000237438.2_17537963_17540893 6 5803 25350 rs2845402 chr22 17542810 17542810 356 299.541 1.08226 2456.19 5.99043e-30 -1.01149 0.000999001 7.76962e-24<br>
		ENSG00000273203.1 chr22 17548711 17548711 + ENSG00000273203.1_17548711_17551565 1 5889 -31323 rs3016112 chr22 17517388 17517388 356 305.346 1.05214 604.311 5.86611e-06 -0.140609 0.00999001 0.0128458<br>
		ENSG00000273442.1 chr22 17561591 17561591 + ENSG00000273442.1_17561591_17562346 1 5911 -8065 rs73149812 chr22 17553526 17553526 356 308.754 1.08518 661.618 0.000244522 0.134045 0.318681 0.304744<br>
		ENSG00000177663.9 chr22 17565844 17565844 + ENSG00000177663.9_17588617_17588658 13 5919 23365 rs879577 chr22 17589209 17589209 356 318.783 1.04364 9233.61 1.09512e-14 -0.564073 0.000999001 1.00321e-09<br>
		ENSG00000183307.3 chr22 17602257 17602257 - ENSG00000183307.3_17597189_17602257 1 5967 14282 rs6518661 chr22 17587975 17587975 356 335.297 0.983505 921.283 1.56789e-07 -0.00776053 0.000999001 0.00038022<br>
		ENSG00000069998.8 chr22 17646177 17646177 - ENSG00000069998.8_17630432_17630635 7 6044 15691 rs1034859 chr22 17630486 17630486 356 300.653 1.15824 3557.61 4.59119e-26 -3.0851 0.000999001 1.38277e-21<br>
		ENSG00000093072.11 chr22 17702879 17702879 - ENSG00000093072.11_17669229_17669617 10 6048 33573 rs2231495 chr22 17669306 17669306 356 293.005 0.951732 2908.66 6.71901e-10 -0.503216 0.001998 0.000102004<br>
		ENSG00000099954.14 chr22 17840837 17840837 + ENSG00000099954.14_18032535_18037850 1 6200 -664900 rs138493943 chr22 17175937 17175937 356 216.381 0.972885 228.553 0.000188795 0.00948545 0.565435 0.576148<br>
		</code>

		<p>
		This file is similar to the one given by a standard permutation pass, but it contains the following differences:
		</p>

		<ul>
			<li>1. The phenotype <b>group</b> ID (here a gene ID)</li>
			<li>2. The chromosome ID of the phenotype group</li>
			<li>3. The start position of the phenotype group</li>
			<li>4. The end position of the phenotype group</li>
			<li>5. The strand orientation of the phenotype group</li>
			<li>6. <b>The top phenotype in the group</b> (here an exon ID)</li>
			<li>7. <b>The total number of phenotypes in the group</b> (i.e. #exons)</li>
			<li>8. The total number of variants tested in <i>cis</i></li>
			<li>9. The distance between the phenotype group and the tested variant (accounting for strand orientation)</li>
			<li>10. The ID of the top variant</li>
			<li>11. The chromosome ID of the top variant</li>
			<li>12. The start position of the top variant</li>
			<li>13. The end position of the top variant</li>
			<li>14. The number of degrees of freedom used to compute the P-values</li>
			<li>15. Dummy</li>
			<li>16. The first parameter value of the fitted beta distribution</li>
			<li>17. The second parameter value of the fitted beta distribution (it also gives the effective number of independent tests in the region)</li>
			<li>18. The nominal P-value of association between the top phenotype and the top variant in <i>cis</i></li>
			<li>19. The corresponding regression slope</li>
			<li>20. The P-value of association adjusted for the number of variants and phenotypes tested in <i>cis</i> given by the direct method (i.e. empirircal P-value)</li>
			<li>21. The P-value of association adjusted for the number of variants and phenotypes tested in <i>cis</i> given by the fitted beta distribution. We strongly recommend to use this adjusted P-value in any downstream analysis</li>
		</ul>

		<p>
		We also implemented an alternative approach to collapse all phenotypes per group using the first principal component of a PCA. To use it, run:
		</p>

		<code>
		QTLtools cis --vcf genotypes.chr22.vcf.gz --bed exons.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --permute 1000 --region chr22:17000000-18000000 <b>--grp-pca1</b> --out permutations.group.txt
		</code>

		<p>
		The output is the same excepted that the 6th column does not contain the top phenotype but instead the variance explained by PC1.
		</p>
		
	</div>
	
	<div class="log"><script type="text/javascript"> print_last_modif_date("$Date: 2015-04-01 16:11:24 +0200 (Wed, 01 Apr 2015) $"); </script></div>
</div>
</body>
</html>
