<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">
	<div class="item"  id="item2">
	<p style="font-size:18px"> <b>This page is outdated, please also see the man page. </b></p>
	</div>
	<div class="item"  id="item2">
		<h1>How to run bamstat?</h1>
		<p>
		The <b>bamstat</b> mode of QTLtools requires two files as input:
		</p>

		<ul>
			<li>A <i>BAM</i> file containing the sequence data of a molecular assay.</li>
			<li>A <i>BED</i> file containing the reference annotation for this molecular assay</li>
		</ul>

		<p>
		To illustrate how it works, we provide the 2 following example files:
		</p>

		<ul>
			<li>A RNA-seq <i>BAM</i> file on chr22 for sample HG00381: <a href="http://jungle.unige.ch/QTLtools_examples/HG00381.chr22.bam" download>BAM</a> / <a href="http://jungle.unige.ch/QTLtools_examples/HG00381.chr22.bam.bai" download>index</a></li>
			<li>An annotation <i>BED</i> file on chr22 derived from GENCODEv19: <a href="http://jungle.unige.ch/QTLtools_examples/gencode.v19.exon.chr22.bed.gz" download>link</a></li>
		</ul>

		<p>
		Then, you can run <b>bamstat</b> on these two files running the command:
		</p>

		<code>
		QTLtools <b>bamstat</b> --bam HG00381.chr22.bam --bed gencode.v19.exon.chr22.bed.gz --filter-mapping-quality 150 --out HG00381.chr22.bamstat.txt
		</code>
		
		<p>
		Note that we use the option <i>--filter-mapping-quality 150</i> since it is the recommended value to remove bad quality reads in a BAM generated with the <a href="http://algorithms.cnag.cat/wiki/The_GEM_library" target='_top'>GEM mapper</a>.
		</p>

		<p>
		This command produces an output file <i>HG00381.chr22.bamstat.txt</i> as follows:
		</p>
		
		<code>
		610860 489276 469678 26771 20179
		</code>
		
		<p>
		The 5 columns give:
		</p>

		<ul>
			<li>A. The total number of sequencing reads in the BAM file (=610,860)</li>
			<li>B. The number of mapped sequencing reads passing the <i>--filter-mapping-quality</i> filter (=489,276)</li>
			<li>C. The number of mapped sequencing reads falling within the annotations specified with <i>--bed</i> (=469,678)</li>
			<li>D. The total number of annotations in the <i>--bed</i> file (=26,771)</li>
			<li>E. The number of annotations covered by at least one sequencing read (=20,179)</li>
		</ul>

		<p>
		From this file, you can then compute the ratios B/A=80%, C/A=77% or E/D=75% to assess the quality of your sequence data.
		</p>
	</div>
	
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
