<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML><HEAD>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>

<TITLE>Man page of QTLtools-trans</TITLE>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>

</HEAD><BODY>
<div class="content">
<div class="item"  id="item2">
<H1>QTLtools-trans</H1><P>
Section: Bioinformatics tools (1)<BR>Updated: 06 May 2020<BR><A HREF="#index">Index</A>
<A HREF="basic_homepage.html">Return to Main Contents</A><HR>

<A NAME="lbAB">&nbsp;</A>
<H2>NAME</H2><P>

QTLtools trans - trans QTL analysis
<A NAME="lbAC">&nbsp;</A>
<H2>SYNOPSIS</H2><P>

<B>QTLtools trans --vcf</B>

<I>[in.vcf</I>|<I>in.vcf.gz</I>|<I>in.bcf</I>|<I>in.bed.gz]</I>

<B>--bed</B>

<I>quantifications.bed.gz</I>

<B>[--nominal</B>

<B>| --permute </B>

<B>| --sample </B>

<I>integer</I>

<B>| --adjust </B>

<I>in.txt</I>]

<B>--out </B>

<I>output.txt</I>

<I>[OPTIONS]</I>

<A NAME="lbAD">&nbsp;</A>
<H2>DESCRIPTION</H2><P>

This mode maps trans (distal) quantitative trait loci (QTLs) that affect the phenotypes, using linear regression. 
The method is detailed in &lt;<A HREF="https://www.nature.com/articles/ncomms15452">https://www.nature.com/articles/ncomms15452</A>&gt;.
We first regress out the provided covariates from the phenotype data, followed by running the linear regression between the phenotype residuals and the genotype.
If <B>--normal</B> and <B>--cov</B> are provided at the same time, then the residuals after the covariate correction are rank normal transformed.
It incorporates an efficient permutation scheme.
You can run a nominal pass (<B>--nominal</B>) listing all genotype-phenotype associations below a certain threshold, a permutation pass (<B>--permute</B> or <B>--sample</B> <I>no_genes_to_sample</I>) to empirically characterize the null distribution of associations, or adjust the nominal p-values based on permutations (<B>--adjust</B>). 
<P>

In the full permutation scheme (<B>--permute</B>) we permute all phenotypes using the same random number sequence to preserve the correlation structure.
By doing so, the only association we actually break in the data is between the genotype and the phenotype data.
Then, we proceed with a standard association scan identical to the one used in the nominal pass.
In practice, we repeat this for 100 permutations of the phenotype data.
Subsequently, we can proceed with FDR correction by ranking all the nominal p-values in ascending order and by counting how many p-values in the permuted data sets are smaller.
This provides an FDR estimate: if we have 500 p-values in the permuted data sets that are smaller than the 100th smallest nominal p-value, we can then assume that the FDR for the 100 first associations is around 5% (=500/(100 Ã— 100)).
<P>

To enable fast screening in trans, we also designed an approximation of the method described just above based on what we already do in cis.
To make it possible, we assume that the phenotypes are independent and normally distributed (which can be enforced with <B>--normal</B>).
The idea is that since all phenotypes are normally distributed, effectively they are the same, and also the cis region removed from each phenotype is so small compared to rest of the genome that its phenotype specific impact is negligible.
Hence the number of and the correlation amongst variants for each phenotype is approximately the same, and each phenotype is approximately the same; thus we can run permutations with a small number of phenotypes rather then all, which drastically decreases the computational burden and the null distribution generated can be applied to all phenotypes.
The implementation draws from the null by permuting some randomly chosen phenotypes, testing for associations with all variants in trans and storing the smallest p-value.
When we repeat this many times (typically 1000), effectively building a null distribution of the strongest associations for a single phenotype.
We then make it continuous by fitting a beta distribution as we do in cis and use it to adjust every nominal p-value coming from the initial pass for the number of variants being tested.
To correct for the number of phenotypes being tested, we estimate FDR as we do in cis; that is from the best adjusted p-values per phenotype (one per phenotype).
This also gives an adjusted p-value threshold that we use to identify all phenotype-variant pairs that are whole-genome significant.
In our experiments, this approach gives similar results to the full permutation scheme both in term of FDR estimates and number of discoveries, while running faster.
<P>

Since linear regressions assumes normally distributed data, we <B>highly recommend</B> using the <B>--normal</B> option to rank normal transform the phenotype quantifications in order to avoid false positive associations due to outliers.
If you are using the approximate permutation scheme (<B>--sample</B>) <B>you MUST use the --normal option or make sure that your phenotypes are normally distributed</B>.  
<A NAME="lbAE">&nbsp;</A>
<H2>OPTIONS</H2><P>

<DL COMPACT>
<DT><B>--vcf [</B><I>in.vcf</I>|<I>in.bcf</I>|<I>in.vcf.gz</I><B>|</B><I>in.bed.gz</I><B>]</B>

<DD>
Genotypes in VCF/BCF format, or another molecular phenotype in BED format.
If there is a DS field in the genotype FORMAT of a variant (dosage of the genotype calculated from genotype probabilities, e.g. after imputation), then this is used as the genotype.
If there is only the GT field in the genotype FORMAT then this is used and it is converted to a dosage.
REQUIRED.
<DT><B>--bed </B><I>quantifications.bed.gz</I>

<DD>
Molecular phenotype quantifications in BED format.
REQUIRED.
<DT><B>--out </B><I>output.txt</I>

<DD>
Output file. 
REQUIRED.
<DT><B>--cov </B><I>covariates.txt</I>

<DD>
Covariates to correct the phenotype data with.
<DT><B>--normal</B>

<DD>
Rank normal transform the phenotype data so that each phenotype is normally distributed.
RECOMMENDED.
<DT><B>--window </B><I>integer</I>

<DD>
Size of the cis window to remove flanking each phenotype's start position.
DEFAULT=5000000.
<DT><B>--threshold </B><I>float</I>

<DD>
P-value threshold below which hits are reported.
Give 1.0 to print everything, which may generate a huge file.
When <B>--adjust</B> is provided, this threshold applies to the adjusted p-values.
DEFAULT=1e-5. 
<DT><B>--bins </B><I>integer</I>

<DD>
Number of bins to use to categorize all p-values above --threshold.
DEFAULT=1000. 
<DT><B>--nominal</B>

<DD>
Calculate the nominal p-value for the genotype-phenotype associations and print out the ones that pass the provided threshold.
Mutually exclusive with --permute, --sample and --adjust.
<DT><B>--permute</B>

<DD>
Permute all phenotypes together, once.
For multiple permutations you need to change the random seed using <B>--seed</B> for each permutation.
Mutually exclusive with --nominal, --sample and --adjust.
<DT><B>--sample </B><I>integer</I>

<DD>
Permute randomly chosen phenotypes <I>integer</I> times.
Mutually exclusive with --nominal, --permute, --adjust, and --chunk.
<DT><B>--adjust </B><I>filename</I>

<DD>
Test and adjust p-values using the null distribution in <I>filename</I>.
Mutually exclusive with --nominal, --permute, and --sample.
<DT><B>--chunk </B><I>integer1</I> <I>integer2</I>

<DD>
For parallelization.
Divide the data into <I>integer2</I> number of chunks and process chunk number <I>integer1</I>.
<B>Minimum number of chunks has to be at least the same number of chromosomes in the --bed file.</B>
<P>
</DL>
<A NAME="lbAF">&nbsp;</A>
<H2>OUTPUT FILES</H2><P>

<DL COMPACT>
<DT><B>.hits.txt.gz </B>

<DD>
Space separated results output file detailing the variant-phenotype pairs that pass the threshold with the following columns:
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The phenotype ID 
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The phenotype chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
Start position of the phenotype
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The variant ID
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The variant chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>6&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The start position of the variant
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>7&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The nominal p-value of the association between the variant and the phenotype.
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>8&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The adjusted p-value of the association between the variant and the phenotype.
Requires <B>--adjust</B>
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>9&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
Correlation coefficient
<BR></TD></TR>
</TABLE>

<P>
<DT><B>.best.txt.gz</B>

<DD>
Space separated output file listing the most significant variant per phenotype.
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The phenotype ID 
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The adjusted p-value of the association between the variant and the phenotype.
Requires <B>--adjust</B>
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The nominal p-value of the association between the variant and the phenotype.
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The variant ID
<BR></TD></TR>
</TABLE>

<DT><B>.bins.txt.gz</B>

<DD>
Space separated output file containing the binning of all hits with a p-value below the specified <B>--threshold</B>.
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The index of the bin
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The lower bound of the correlation coefficient for this bin
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The upper bound of the correlation coefficient for this bin
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The upper bound of the p-value for this bin
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The lower bound of the p-value for this bin
<BR></TD></TR>
</TABLE>

<P>
</DL>
<A NAME="lbAG">&nbsp;</A>
<H2>FULL PERMUTATION ANALYSIS EXAMPLE</H2><P>

<DL COMPACT>
<DT>1
Run a nominal analysis, rank normal transforming the phenotypes and outputting all associations with a p-value below 1e-5:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools trans --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --nominal --normal --out trans.nominal
</CODE style="font-size:1em;">
<P>
<DT>2
Run a full permutation analysis with 100 jobs on a compute cluster, run the following <B>making sure that you change the seed for each permutation iteration</B> (qsub needs to be changed to the job submission system used [bsub, psub, etc...])
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
for j in $(seq 1 100); do
<BR>
<TD>&nbsp;&nbsp;&nbsp;&nbsp;</TD>echo &quot;QTLtools trans --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --permute --normal --out trans.perm$j.txt --seed $j&quot; | qsub
<BR>
done
</CODE>
<P>
</DL>
<A NAME="lbAH">&nbsp;</A>
<H2>APPROXIMATE PERMUTATION ANALYSIS EXAMPLE</H2><P>

<DL COMPACT>
<DT>1
Build the null distribution randomly selecting 1000 phenotypes, and rank normal transforming the phenotypes:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools trans --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --sample 1000 --normal --out trans.sample
</CODE>
<P>
<DT>2
Run the nominal pass adjusting the p-values with the given null distribution, rank normal transforming the phenotypes, and printing out associations with an adjusted p-value less than 0.1:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools trans --vcf genotypes.chr22.vcf.gz --bed genes.simulated.chr22.bed.gz --adjust trans.sample.best.txt.gz --threshold 0.1 --normal --out trans.adjust
</CODE>
<P>
</DL>
<A NAME="lbAI">&nbsp;</A>
<H2>SEE ALSO</H2><P>

<I><A HREF="QTLtools.1.html">QTLtools</A></I>(1)











<A NAME="lbAJ">&nbsp;</A>
<H2>BUGS</H2><P>

<DL COMPACT>
<DT>o
Versions up to and including 1.2, suffer from a bug in reading missing genotypes in VCF/BCF files. 
This bug affects variants with a DS field in their genotype's FORMAT and have a missing genotype (DS field is .) in one of the samples, in which case genotypes for all the samples are set to missing, effectively removing this variant from the analyses.
</DL>
<P>

Please submit bugs to &lt;<A HREF="https://github.com/qtltools/qtltools">https://github.com/qtltools/qtltools</A>&gt;
<A NAME="lbAK">&nbsp;</A>
<H2>CITATION</H2><P>

Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis. <I>Nat Commun</I> <B>8</B>, 15452 (2017). 
&lt;<A HREF="https://doi.org/10.1038/ncomms15452">https://doi.org/10.1038/ncomms15452</A>&gt;
<A NAME="lbAL">&nbsp;</A>
<H2>AUTHORS</H2><P>

Halit Ongen (<A HREF="mailto:halitongen@gmail.com">halitongen@gmail.com</A>), Olivier Delaneau (<A HREF="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</A>)
<P>

<HR>
<A NAME="index">&nbsp;</A><H2>Index</H2><P>
<DL>
<DT><A HREF="#lbAB">NAME</A><DD>
<DT><A HREF="#lbAC">SYNOPSIS</A><DD>
<DT><A HREF="#lbAD">DESCRIPTION</A><DD>
<DT><A HREF="#lbAE">OPTIONS</A><DD>
<DT><A HREF="#lbAF">OUTPUT FILES</A><DD>
<DT><A HREF="#lbAG">FULL PERMUTATION ANALYSIS EXAMPLE</A><DD>
<DT><A HREF="#lbAH">APPROXIMATE PERMUTATION ANALYSIS EXAMPLE</A><DD>
<DT><A HREF="#lbAI">SEE ALSO</A><DD>
<DT><A HREF="#lbAJ">BUGS</A><DD>
<DT><A HREF="#lbAK">CITATION</A><DD>
<DT><A HREF="#lbAL">AUTHORS</A><DD>
</DL>
<HR>
This document was created by
<A HREF="/cgi-bin/man/man2html">man2html</A>,
using the manual pages.<BR>
Time: 14:11:25 GMT, August 24, 2020
</div>
</div>
</BODY>
</HTML>
