<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML><HEAD>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>

<TITLE>Man page of QTLtools-rtc</TITLE>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>

</HEAD><BODY>
<div class="content">
<div class="item"  id="item2">
<H1>QTLtools-rtc</H1><P>
Section: Bioinformatics tools (1)<BR>Updated: 06 May 2020<BR><A HREF="#index">Index</A>
<A HREF="basic_homepage.html">Return to Main Contents</A><HR>

<A NAME="lbAB">&nbsp;</A>
<H2>NAME</H2><P>

QTLtools rtc - Regulatory Trait Concordance score analysis 
<A NAME="lbAC">&nbsp;</A>
<H2>SYNOPSIS</H2><P>

<B>QTLtools rtc  --vcf</B>

<I>[in.vcf</I>|<I>in.vcf.gz</I>|<I>in.bcf</I>|<I>in.bed.gz]</I>

<B>--bed</B>

<I>quantifications.bed.gz</I>

<B>--hotspots</B>

<I>hotspots_b37_hg19.bed</I>

<B>[--gwas-cis | --gwas-trans | --mergeQTL-cis | --mergeQTL-trans]</B>

<I>variants_external.txt qtls_in_this_dataset.txt</I>

<B>--out </B>

<I>output.txt</I>

<I>[OPTIONS]</I>

<A NAME="lbAD">&nbsp;</A>
<H2>DESCRIPTION</H2><P>

The RTC algorithm assesses the likelihood of a shared functional effect between a GWAS variant and an molQTL by quantifying the change in the statistical significance of the molQTL after correcting the molQTL phenotype for the genetic effect of the GWAS variant and comparing its correction impact to that of all other SNPs in the interval.
The method is detailed in &lt;<A HREF="https://www.nature.com/articles/ng.3981">https://www.nature.com/articles/ng.3981</A>&gt;.
When assessing tissue specificity of molQTLs we use the same method, however in that case the GWAS variant becomes an molQTL in a different tissue.
The RTC method is as follows: for a GWAS variant falling into the same region flanked by recombination hotspots (coldspot) with an molQTL, with N number of variants in a given coldspot:
<DL COMPACT>
<DT>1
Correct the phenotype for each of the variants in the region separately by linear regression, resulting in N number of pseudo-phenotypes (residuals).
<DT>2
Redo the molQTL variant association with all of these pseudo-phenotypes.
<DT>3
Sort (decreasing) the resulting p-values and find the rank of the molQTL to GWAS-pseudo-phenotype among all molQTL to pseudo-phenotype associations.
<DT>4
RTC = (N - Rank of GWAS) / N
</DL>
<P>

This results in the RTC score which ranges from 0 to 1 where higher values indicate a more likely shared functional effect between the GWAS and the molQTL variants.
An RTC score greater than or equal to 0.9 is considered a shared functional effect.
If there are multiple independent molQTLs for a given phenotype, RTC for each independent molQTL is assessed after correcting the phenotype with all the other molQTL variants for that phenotype.
This correction is done using linear regression and taking the residuals after regressing the phenotype with the other molQTLs.
<P>

In order to convert RTC score into a probability of sharing, we employ two simulations per coldspot region, H0 and H1.
The H0 scenario is when two variants in a coldspot are tagging different functional effects.
For a coldspot that harbours colocalized GWAS and molQTL variants, we pick two random hidden causal variants. 
We then find two variants (GWAS and molQTL) that are linked (default r-squared â‰¥ 0.5) to the hidden causal variants.
We generate a pseudo phenotype for molQTL based on the slope and intercept of the observed molQTL and randomly distributed residuals of the observed molQTL.
Subsequently we rerun the RTC analysis with this new pseudo-phenotype and using the GWAS and molQTL variants.
<P>

The H1 scenario is when the two variants are tagging the same functional variant. 
The scheme here is exactly the same as the H0 scheme, except there is only one hidden causal variant and both GWAS and molQTL variants are randomly selected from variants that are linked to the same hidden causal variant. 
<A NAME="lbAE">&nbsp;</A>
<H2>OPTIONS</H2><P>

<DL COMPACT>
<DT><B>--vcf [</B><I>in.vcf</I>|<I>in.bcf</I>|<I>in.vcf.gz</I><B>|</B><I>in.bed.gz</I><B>]</B>

<DD>
Genotypes in VCF/BCF format, or another molecular phenotype in BED format.
If there is a DS field in the genotype FORMAT of a variant (dosage of the genotype calculated from genotype probabilities, e.g. after imputation), then this is used as the genotype.
If there is only the GT field in the genotype FORMAT then this is used and it is converted to a dosage.
REQUIRED.
<DT><B>--bed </B><I>quantifications.bed.gz</I>

<DD>
Molecular phenotype quantifications in BED format.
REQUIRED.
<DT><B>--out </B><I>output.txt</I>

<DD>
Output file. 
REQUIRED.
<DT><B>--hotspots </B><I>recombination_hotspots.bed</I>

<DD>
Recombination hotspots in BED format.
REQUIRED.
<DT><B>--cov </B><I>covariates.txt</I>

<DD>
Covariates to correct the phenotype data with.
<DT><B>--stats-vcf [</B><I>in.vcf</I>|<I>in.bcf</I>|<I>in.vcf.gz</I><B>]</B>

<DD>
Calculate D' and r-squared from this file.
Defaults to the --vcf file.
Needs to have phased genotypes for D' calculations.
<DT><B>--stats-vcf-include-samples </B><I>samples.txt</I>

<DD>
Samples to include from the --stats-vcf file.
One sample ID per line.
<DT><B>--stats-vcf-exclude-samples </B><I>samples.txt</I>

<DD>
Samples to exclude from the --stats-vcf file.
One sample ID per line.
<DT><B>--normal</B>

<DD>
Rank normal transform the phenotype data so that each phenotype is normally distributed.
RECOMMENDED.
<DT><B>--conditional</B>

<DD>
molQTLs contain independent signals so execute the conditional analysis.
<DT><B>--debug</B>

<DD>
Print out debugging info to stderr.
DON'T USE.
<DT><B>--warnings</B>

<DD>
Print all encountered individual warnings to stdout.
<DT><B>--header</B>

<DD>
Add a header to the output file when --chunk or --region is provided.
<DT><B>--individual-Dprime</B>

<DD>
Calculate D' on an individual variant basis.
If not provided D' will not be calculated after first unphased genotype is encountered.
<DT><B>--mem-est</B>

<DD>
Estimate memory usage and exit.
<DT><B>--mem [</B><I>0</I>|<I>1</I>|<I>2</I>|<I>3</I><B>]</B>

<DD>
Keep results of calculations that may be used multiple times in memory. 0 = nothing in mem, 1 = only basic, 2 = all in mem but clean after unlikely to be reused, 3 = all in mem no cleaning.
DEFAULT=0. 
RECOMMENDED=2.
<DT><B>--window </B><I>integer</I>

<DD>
Size of the cis window flanking each phenotype's start position.
DEFAULT=1000000. 
RECOMMENDED=1000000.
<DT><B>--sample </B><I>integer</I>

<DD>
Number of simulated RTC values to try to achieve for each coldspot, for converting RTC to a probability.
At each iteration we try to pick a unique combination of variants, thus the actual number of sample iterations may be less than this value, due to the number variants in a region.
If you want to run this analysis, please provide at least 100.
DEFAULT=0.
<DT><B>--max-sample </B><I>integer</I>

<DD>
Max number of sample iterations trying to reach --sample before quitting. 
Provide the actual number not the multiplier.
DEFAULT=--sample * 50.
<DT><B>--R2-threshold </B><I>float</I>

<DD>
The minimum r-squared required when picking a variant that is linked to the hidden causal variant(s) when running simulations using --sample. 
DEFAULT=0.5.
<DT><B>--D-prime-threshold </B><I>float</I>

<DD>
If the pairs of variants fall into different coldspots and have a D' greater than this, the RTC calculation is extended to multiple coldspot regions including both variants. 
Assumes D' can be calculated.
DEFAULT=OFF.
NOT RECOMMENDED.
<DT><B>--grp-best</B>

<DD>
Correct for multiple phenotypes within a phenotype group.
<DT><B>--pheno-col </B><I>integer</I>

<DD>
1-based phenotype id column number.
DEFAULT=1 or 5 when --grp-best
<DT><B>--geno-col </B><I>integer</I>

<DD>
1-based genotype id column number.
DEFAULT=8 or 10 when --grp-best
<DT><B>--grp-col </B><I>integer</I>

<DD>
1-based phenotype group id column number.
Only relevant if <B>--grp-best</B> is in effect.
DEFAULT=1
<DT><B>--rank-col </B><I>integer</I>

<DD>
1-based conditional analysis rank column number.
Only relevant if <B>--conditional</B> is in effect.
DEFAULT=12 or 14 when --grp-best
<DT><B>--best-col </B><I>integer</I>

<DD>
1-based phenotype column number
Only relevant if <B>--conditional</B> is in effect.
DEFAULT=21 or 23 when --grp-best
<DT><B>--gwas-cis </B><I>variants_external.txt qtls_in_this_dataset.txt</I>

<DD>
Run RTC for GWAS and cis-molQTL colocalization analysis.
Takes two file names as arguments. 
The first is the file with GWAS variants of interest with one variant ID per line.
These should match the variants IDs in the <B>--vcf</B> file.
The second is the QTLtools output for the cis run that was ran using the same <B>--vcf</B>, <B>--bed</B>, and <B>--cov</B> files. 
REQUIRED unless (and mutually exclusive with) <B>--gwas-trans</B>, <B>--mergeQTL-cis</B>, <B>--mergeQTL-trans</B>.
<DT><B>--gwas-trans </B><I>variants_external.txt qtls_in_this_dataset.txt</I>

<DD>
Run RTC for GWAS and trans-molQTL colocalization analysis.
Takes two file names as arguments. 
The first is the file with GWAS variants of interest with one variant ID per line.
These should match the variants IDs in the <B>--vcf</B> file.
The second is the QTLtools output for the trans run that was ran using the same <B>--vcf</B>, <B>--bed</B>, and <B>--cov</B> files.
You will need to adjust <B>*-col</B> options. 
REQUIRED unless (and mutually exclusive with) <B>--gwas-cis</B>, <B>--mergeQTL-cis</B>, <B>--mergeQTL-trans</B>.
<DT><B>--mergeQTL-cis </B><I>variants_external.txt qtls_in_this_dataset.txt</I>

<DD>
Run RTC for cis-molQTL and cis-molQTL colocalization analysis.
Takes two file names as arguments. 
The first is the file with cis-molQTL variants of interest discovered in a different dataset, e.g. different tissue, with one variant ID per line.
These should match the variants IDs in the <B>--vcf</B> file.
The second is the QTLtools output for the cis run that was ran using the same <B>--vcf</B>, <B>--bed</B>, and <B>--cov</B> files. 
REQUIRED unless (and mutually exclusive with) <B>--gwas-trans</B>, <B>--mergeQTL-cis</B>, <B>--mergeQTL-trans</B>.
<DT><B>--mergeQTL-trans </B><I>variants_external.txt qtls_in_this_dataset.txt</I>

<DD>
Run RTC for trans-molQTL and trans-molQTL colocalization analysis.
Takes two file names as arguments. 
The first is the file with trans-molQTL variants of interest discovered in a different dataset, e.g. different tissue, with one variant ID per line.
These should match the variants IDs in the <B>--vcf</B> file.
The second is the QTLtools output for the trans run that was ran using the same <B>--vcf</B>, <B>--bed</B>, and <B>--cov</B> files. 
You will need to adjust <B>*-col</B> options. 
REQUIRED unless (and mutually exclusive with) <B>--gwas-trans</B>, <B>--gwas-cis</B>, <B>--mergeQTL-cis</B>.
<DT><B>--chunk </B><I>integer1</I> <I>integer2</I>

<DD>
For parallelization.
Divide the data into <I>integer2</I> number of chunks and process chunk number <I>integer1</I>.
Chunk 0 will print a header.
Mutually exclusive with --region.
<B>Minimum number of chunks has to be at least the same number of chromosomes in the --bed file.</B>
<DT><B>--region </B><I>chr:start-end</I>

<DD>
Genomic region to be processed.
E.g. chr4:12334456-16334456, or chr5.
Mutually exclusive with --chunk.
<P>
</DL>
<A NAME="lbAF">&nbsp;</A>
<H2>OUTPUT FILE</H2><P>

<DL COMPACT>
<DT><B>--out output file</B>

<DD>
Space separated output file with the following columns.
Columns after the 22nd are only printed if <B>--sample</B> is provided.
We recommend including chunk 0 to print out a header in order to avoid confusion.
<P>
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>other_variant</TD><TD>
The variant ID that is external to this dataset, could be the GWAS variant or another molQTL 
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>our_variant</TD><TD>
The molQTL variant ID that is internal to this dataset
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>phenotype</TD><TD>
The phenotype ID
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>phenotype_group</TD><TD>
The phenotype group ID
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>other_variant_chr</TD><TD>
The external variant's chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>6&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>other_variant_start</TD><TD>
The external variant's start position
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>7&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>other_variant_rank</TD><TD>
Rank of the external variant.
Only relevant if the external variants are part of an conditional analysis
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>8&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>our_variant_chr</TD><TD>
The internal variant's chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>9&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>our_variant_start</TD><TD>
The internal variant's start position
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>10&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>our_variant_rank</TD><TD>
Rank of the internal variant.
Only relevant if the internal variants are part of an conditional analysis
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>11&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>phenotype_chr</TD><TD>
The phenotype's chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>12&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>phenotype_start</TD><TD>
The start position of the phenotype
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>13&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>distance_between_variants</TD><TD>
The distance between the two variants
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>14&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>distance_between_other_variant_and_pheno</TD><TD>
The distance between the external variant and the phenotype
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>15&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>other_variant_region_index</TD><TD>
The region index of the external variant
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>16&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>our_variant_region_index</TD><TD>
The region index of the internal variant
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>17&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>region_start</TD><TD>
The start position of the region
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>18&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>region_end</TD><TD>
The end position of the region
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>19&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>variant_count_in_region</TD><TD>
The number of variants in the region
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>20&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>RTC</TD><TD>
The RTC score
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>21&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>D'</TD><TD>
The D' of the two variants.
Only calculated if there are phased genotypes
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>22&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>r^2</TD><TD>
The r squared of the two variants
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>22&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>p_value</TD><TD>
The p-value of the RTC score
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>23&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>unique_picks_H0</TD><TD>
The number of unique combinations of variants in the H0 simulations
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>24&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>unique_picks_H1</TD><TD>
The number of unique combinations of variants in the H1 simulations
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>25&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>rtc_bin_start</TD><TD>
Lower bound of the RTC bin, based on the observed RTC score
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>26&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>rtc_bin_end</TD><TD>
Upper bound of the RTC bin, based on the observed RTC score
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>27&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>rtc_bin_H0_proportion</TD><TD>
The proportion of H0 simulated values that are between rtc_bin_start and rtc_bin_end
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>28&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>rtc_bin_H1_proportion</TD><TD>
The proportion of H1 simulated values that are between rtc_bin_start and rtc_bin_end
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>29&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>median_r^2</TD><TD>
The median r-squared in the region
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>30&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>median_H0</TD><TD>
The median RTC score in the H0 simulations
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>31&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>median_H1</TD><TD>
The median RTC score in the H1 simulations
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>32&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>H0</TD><TD>
The RTC scores observed in the H0 simulations
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>33&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>H1</TD><TD>
The RTC scores observed in the H1 simulations
<BR></TD></TR>
</TABLE>

<P>
</DL>
<A NAME="lbAG">&nbsp;</A>
<H2>EXAMPLES</H2><P>

<DL COMPACT>
<DT>o
Run RTC with GWAS variants and cis-eQTLs correcting for technical covariates and rank normal transforming the phenotype:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools rtc --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --hotspot hotspots_b37_hg19.bed --gwas-cis GWAS.b37.txt permutations_all.significant.txt --normal --out rtc_results.txt 
</CODE>
<P>
<DT>o
RTC with GWAS variants and cis-eQTLs and simulations, correcting for technical covariates, rank normal transforming the phenotype, and running conditional analysis while keeping data in memory.
To facilitate parallelization on compute cluster, we developed an option to run the analysis into chunks of molecular phenotypes.
For instance, to run analysis on chunk 12 when splitting the example data set into 20 chunks, run:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools rtc --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --hotspot hotspots_b37_hg19.bed --gwas-cis GWAS.b37.txt conditional_all.significant.txt --normal --conditional --mem 2 --chunk 12 20 --sample 200 --out rtc_results_12_20.txt  
</CODE>
<P>
<DT>o
If you want to submit the whole analysis with 20 jobs on a compute cluster, just run (qsub needs to be changed to the job submission system used [bsub, psub, etc...]):
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
for j in $(seq 0 20); do
<BR>

<TT>&nbsp;&nbsp;&nbsp;&nbsp;</TT> echo &quot;QTLtools rtc --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --cov genes.covariates.pc50.txt.gz --hotspot hotspots_b37_hg19.bed --gwas-cis GWAS.b37.txt conditional_all.significant.txt --normal --conditional --mem 2 --chunk $j 20 --sample 200 --out rtc_results_$j_20.txt&quot; | qsub
<BR>

done
</CODE>
<P>
</DL>
<A NAME="lbAH">&nbsp;</A>
<H2>SEE ALSO</H2><P>

<I><A HREF="QTLtools.1.html">QTLtools</A></I>(1)










<A NAME="lbAI">&nbsp;</A>
<H2>BUGS</H2><P>

<P>
Please submit bugs to &lt;<A HREF="https://github.com/qtltools/qtltools">https://github.com/qtltools/qtltools</A>&gt;
<A NAME="lbAJ">&nbsp;</A>
<H2>CITATION</H2><P>

Ongen H, Brown AA, Delaneau O, et al. Estimating the causal tissues for complex traits and diseases. <I>Nat Genet</I>. 2017;<B>49</B>(12):1676-1683. doi:10.1038/ng.3981
&lt;<A HREF="https://doi.org/10.1038/ng.3981">https://doi.org/10.1038/ng.3981</A>&gt;
<A NAME="lbAK">&nbsp;</A>
<H2>AUTHORS</H2><P>

Halit Ongen (<A HREF="mailto:halitongen@gmail.com">halitongen@gmail.com</A>), Olivier Delaneau (<A HREF="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</A>)
<P>

<HR>
<A NAME="index">&nbsp;</A><H2>Index</H2><P>
<DL>
<DT><A HREF="#lbAB">NAME</A><DD>
<DT><A HREF="#lbAC">SYNOPSIS</A><DD>
<DT><A HREF="#lbAD">DESCRIPTION</A><DD>
<DT><A HREF="#lbAE">OPTIONS</A><DD>
<DT><A HREF="#lbAF">OUTPUT FILE</A><DD>
<DT><A HREF="#lbAG">EXAMPLES</A><DD>
<DT><A HREF="#lbAH">SEE ALSO</A><DD>
<DT><A HREF="#lbAI">BUGS</A><DD>
<DT><A HREF="#lbAJ">CITATION</A><DD>
<DT><A HREF="#lbAK">AUTHORS</A><DD>
</DL>
<HR>
This document was created by
<A HREF="/cgi-bin/man/man2html">man2html</A>,
using the manual pages.<BR>
Time: 14:11:25 GMT, August 24, 2020
</div>
</div>
</BODY>
</HTML>
