<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML><HEAD>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>

<TITLE>Man page of QTLtools-quan</TITLE>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>

</HEAD><BODY>
<div class="content">
<div class="item"  id="item2">
<H1>QTLtools-quan</H1><P>
Section: Bioinformatics tools (1)<BR>Updated: 06 May 2020<BR><A HREF="#index">Index</A>
<A HREF="basic_homepage.html">Return to Main Contents</A><HR>

<A NAME="lbAB">&nbsp;</A>
<H2>NAME</H2><P>

QTLtools quan - Quantify gene and exon expression from RNA-seq
<A NAME="lbAC">&nbsp;</A>
<H2>SYNOPSIS</H2><P>

<B>QTLtools quan --bam</B>

<I>[in.sam</I>|<I>in.bam</I>|<I>in.cram]</I>

<B>--gtf</B>

<I>gene_annotation.gtf</I>

<B>--out-prefix </B>

<I>output</I>

<I>[OPTIONS]</I>

<A NAME="lbAD">&nbsp;</A>
<H2>DESCRIPTION</H2><P>

This mode quantifies the expression of genes and exons in the provided <B>--gtf</B> file using the RNA-seq reads in the <B>--bam</B> file.
The method counts the number of reads overlapping the exons in the <B>--gtf</B> file.
Firstly all exons of a gene are converted into meta-exons where overlapping exons are merged into a single exon encompassing all the overlapping exons.  
Any overlap between the read and the exon is considered a match, that is a read is not required to be in between start and end positions of an exon to count towards that exon's quantification.
Split reads aligning to multiple exons contribute to each exon it overlaps with based on the fraction of the read that overlaps with a given exon.
Thus split reads contribute less than a single count to each of the overlapping exons.
Reads aligning to multiple exons (i.e. overlapping exons of multiple genes) count towards the quantification of all the exons that it overlaps with.
If the <B>--bam</B> file contains paired-end reads and if there are cases where the two mate pairs overlap with each other (i.e. have an insert size &lt; 0), then each of these reads contribute less then a single count towards the quantifications unless <B>--no-merge</B> is provided.
The following diagram, with two genes with overlapping exons and one paired-end read where both mate pairs are split reads and overlap with each other, illustrates how the quantification works:
<P>

<PRE style="font-size:1.3em;">
             x             x
            / \           / \
 +---------+   +---------+   +---------+
 | Exon1|1 |   | Exon1|2 |   | Exon1|3 |      Gene1
 +---------+   +---------+   +---------+
                           x
                          / \
            +------------+   +-------------+
            |  Exon2|1   |   |   Exon2|2   |  Gene2
            +------------+   +-------------+
                              x
                             / \
               +------------+   +----+        RNAseq Read Mate1
                  
               |--a-||-b-||c|   |-d--||--e-|
                              x
                             / \               
                     +------+   +----------+  RNAseq Read Mate2

  Left Mate1  = ((b * 0.5) + a) / (a + b + d)
  Right Mate1 = (d * 0.5)/(a + b + d)
  Left Mate2  = (b * 0.5)/(b + d + e)
  Right Mate2 = ((d * 0.5) + e)/(b + d + e)

  Exon1|2 = Left Mate1 + Left Mate2
  Exon1|3 = Right Mate1 + Right Mate2
  Exon2|1 = Left Mate1 + Left Mate2
  Exon2|2 = Right Mate2 + Right Mate2

  Gene1 = Exon1|2 + Exon1|3
  Gene2 = Exon2|1 + Exon2|2
</PRE>

<P><BR>
<B>The quan mode in version 1.2 and above is not compatible with the quantifications generated by the previous versions. This due to bug fixes and slight adjustments to the way we quantify. DO NOT MIX QUANTIFICATIONS GENERATED BY EARLIER VERSIONS OF QTLTOOLS WITH QUANTIFICATIONS FROM VERSION 1.2 AND ABOVE AS THIS WILL CREATE A BIAS IN YOUR DATASET.</B>

<A NAME="lbAE">&nbsp;</A>
<H2>OPTIONS</H2><P>

<DL COMPACT>
<DT><B>--gtf </B><I>gene_annotation.gtf</I>

<DD>
Gene annotations in GTF format.
These can be obtained from &lt;<A HREF="https://www.gencodegenes.org/">https://www.gencodegenes.org/</A>&gt;.
REQUIRED.
<DT><B>--bam [</B><I>in.bam</I>|<I>in.sam</I>|<I>in.cram</I><B>]</B>

<DD>
Sequence data in BAM/SAM/CRAM format sorted by chromosome and then position.
One sample per BAM file.
REQUIRED.
<DT><B>--out-prefix </B><I>output</I>

<DD>
Output prefix. 
REQUIRED.
<DT><B>--sample </B><I>sample_name</I>

<DD>
The sample name of the BAM file.
If not provided the sample name will be taken as the BAM file path.
<DT><B>--rpkm</B>

<DD>
Output RPKM values.
<DT><B>--tpm</B>

<DD>
Output TPM values.
<DT><B>--xxhash</B>

<DD>
Rather than using the GTF file name to generate unique hash for the options used, use the hash of the GTF file.
<DT><B>--no-hash</B>

<DD>
Do not include a hash signifying the options used in the quantification in the output file names.
NOT RECOMMENDED.
<DT><B>--gene-types </B><I>gene_type</I> ...

<DD>
Only quantify these gene types.
Requires gene_type attribute in GTF. 
It will also use transcript_type if present.
<DT><B>--filter-mapping-quality </B><I>integer</I>

<DD>
Minimum mapping quality for a read or read pair to be considered. 
Set this to only include uniquely mapped reads.
DEFAULT=10.
<DT><B>--filter-mismatch </B><I>integer|float</I>

<DD>
Maximum mismatches allowed in a read. 
If between 0 and 1 taken as the fraction of read length. 
<B>Requires NM attribute in the BAM file.</B>
DEFAULT=OFF.
<DT><B>--filter-mismatch-total </B><I>integer|float</I>

<DD>
Maximum total mismatches allowed in paired-reads. 
If between 0 and 1 taken as the fraction of combined read length. 
<B>Requires NM attribute in the BAM file.</B>
DEFAULT=OFF.
<DT><B>--filter-min-exon </B><I>integer</I>

<DD>
Minimum length of an exon for it to be quantified.
Exons smaller than this will not be printed out in the exon quantifications, but will still count towards gene quantifications.
DEFAULT=0.
<DT><B>--filter-remove-duplicates</B>

<DD>
Remove duplicate sequencing reads,as indicated by the aligner, in the process.
NOT RECOMMENDED.
<DT><B>--filter-failed-qc</B>

<DD>
Remove fastq reads that fail sequencing QC as indicated by the sequencer.
<DT><B>--check-proper-pairing</B>

<DD>
If provided only properly paired reads according to the aligner that are in correct orientation will be considered.
Otherwise all pairs in correct orientation will be considered.
<DT><B>--check-consistency</B>

<DD>
If provided checks the consistency of split reads with annotation, rather than pure overlap of one of the blocks of the split read.
<DT><B>--no-merge</B>

<DD>
If provided overlapping mate pairs will not be merged. 
Default behavior is to merge overlapping mate pairs based on the amount of overlap, such that each mate pair counts for less than 1 read.
<DT><B>--legacy-options</B>

<DD>
Exactly replicate Dermitzakis lab original quantification script.
DO NOT USE.
<DT><B>--region </B><I>chr:start-end</I>

<DD>
Genomic region to be processed.
E.g. chr4:12334456-16334456, or chr5.
<P>
</DL>
<A NAME="lbAF">&nbsp;</A>
<H2>OUTPUT FILES</H2><P>

Unless <B>--no-hash</B> is provided, all output files will include a hash value corresponding to combination of the specific options used.
This is given so that one does not merge quantifications from samples that were quantified differently, which would create a bias in the dataset.
<DL COMPACT>
<DT><B>.gene.count.bed .exon.count.bed .gene.rpkm.bed .exon.rpkm.bed .gene.tpm.bed .exon.tpm.bed</B>

<DD>
These are the quantification results files with the following columns:
<P>
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>chr</TD><TD>
Phenotype's chromosome
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>start</TD><TD>
Phenotype's start position (0-based)
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>end</TD><TD>
Phenotype's end position (1-based)
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>gene|exon</TD><TD>
The gene or exon ID.
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>info|geneID</TD><TD>
Information about the gene or the gene ID of the exon.
The gene info is separated by semicolons, and L=gene length, T=gene type, R=gene positions, N=gene name
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>6&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>strand</TD><TD>
Phenotype's strand
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>7&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>sample_name</TD><TD>
The sample name of the BAM file
<BR></TD></TR>
</TABLE>

<P>
<DT><B>.stats</B>

<DD>
Details the statistics of the quantification, with the following rows:
<P>
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_secondary_alignments_(does_not_count_towards_total_reads)</TD><TD>
Number of secondary alignments
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_reads</TD><TD>
Number of reads in the BAM file
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_unmapped</TD><TD>
Number of unmapped reads
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_failqc</TD><TD>
Number of reads with the failed QC tag
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_duplicate</TD><TD>
Number of duplicate reads
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>6&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_mapQ_less_than_X</TD><TD>
Number of reads below the mapping quality threshold X
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>7&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_notpaired</TD><TD>
Number of pairs that were not in the correct orientation or were not properly paired
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>8&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_mismatches_greater_than_X_Y</TD><TD>
Number of reads failing the mismatches per read, X, and mismatches total filters, Y
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>9&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>filtered_unmatched_mate_pairs</TD><TD>
Number of reads where there was a paired-read with a missing mate
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>10&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_good</TD><TD>
Number of reads that passed all filters
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>11&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_exonic</TD><TD>
Number of reads that aligned to exons and passed all filters
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>12&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_exonic_multi_counting</TD><TD>
Number of reads that aligned to exons when we count reads that align to multiple exons multiple times
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>13&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_merged_reads</TD><TD>
Number of reads where the mate pairs were overlapping and thus were merged
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>14&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>total_exonic_multi_counting_after_merge_(used_for_rpkm)</TD><TD>
Number of reads that aligned to exons when we merge overlapping mate pairs
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>15&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>good_over_total</TD><TD>
Number of good reads over the total number of reads
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>16&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>exonic_over_total</TD><TD>
Number of exonic reads over the total number of reads
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>17&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>exonic_over_good</TD><TD>
Number of exonic reads over the number of good reads
<BR></TD></TR>
</TABLE>

<P>
</DL>
<A NAME="lbAG">&nbsp;</A>
<H2>EXAMPLE</H2><P>

<DL COMPACT>
<DT>o
Quantifying a sample mapped with GEM, outputting TPM and RPKM values, and taking the hash of the GTF file:
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools quan --bam HG00381.chr22.bam --gtf gencode.v19.annotation.chr22.gtf.gz --out-prefix HG00381 --sample HG00381 --rpkm --tpm --xxhash --filter-mismatch-total 8 --filter-mapping-quality 150
</CODE>
</DL>
<A NAME="lbAH">&nbsp;</A>
<H2>SEE ALSO</H2><P>

<I><A HREF="QTLtools.1.html">QTLtools</A></I>(1)














<A NAME="lbAI">&nbsp;</A>
<H2>BUGS</H2><P>

Please submit bugs to &lt;<A HREF="https://github.com/qtltools/qtltools">https://github.com/qtltools/qtltools</A>&gt;
<A NAME="lbAJ">&nbsp;</A>
<H2>CITATION</H2><P>

Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis. <I>Nat Commun</I> <B>8</B>, 15452 (2017). 
&lt;<A HREF="https://doi.org/10.1038/ncomms15452">https://doi.org/10.1038/ncomms15452</A>&gt;
<A NAME="lbAK">&nbsp;</A>
<H2>AUTHORS</H2><P>

Halit Ongen (<A HREF="mailto:halitongen@gmail.com">halitongen@gmail.com</A>), Olivier Delaneau (<A HREF="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</A>)
<P>

<HR>
<A NAME="index">&nbsp;</A><H2>Index</H2><P>
<DL>
<DT><A HREF="#lbAB">NAME</A><DD>
<DT><A HREF="#lbAC">SYNOPSIS</A><DD>
<DT><A HREF="#lbAD">DESCRIPTION</A><DD>
<DT><A HREF="#lbAE">OPTIONS</A><DD>
<DT><A HREF="#lbAF">OUTPUT FILES</A><DD>
<DT><A HREF="#lbAG">EXAMPLE</A><DD>
<DT><A HREF="#lbAH">SEE ALSO</A><DD>
<DT><A HREF="#lbAI">BUGS</A><DD>
<DT><A HREF="#lbAJ">CITATION</A><DD>
<DT><A HREF="#lbAK">AUTHORS</A><DD>
</DL>
<HR>
This document was created by
<A HREF="/cgi-bin/man/man2html">man2html</A>,
using the manual pages.<BR>
Time: 14:11:25 GMT, August 24, 2020
</div>
</div>
</BODY>
</HTML>
