<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML><HEAD>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-176212697-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-176212697-1');
</script>

<TITLE>Man page of QTLtools-bamstat</TITLE>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>

</HEAD><BODY>
<div class="content">
<div class="item"  id="item2">
<H1>QTLtools-bamstat</H1><P>
Section: Bioinformatics tools (1)<BR>Updated: 06 May 2020<BR><A HREF="#index">Index</A>
<A HREF="basic_homepage.html">Return to Main Contents</A><HR>

<A NAME="lbAB">&nbsp;</A>
<H2>NAME</H2><P>

QTLtools bamstat - Calculate stats of overlap between an RNAseq BAM file and an annotation
<A NAME="lbAC">&nbsp;</A>
<H2>SYNOPSIS</H2><P>

<B>QTLtools bamstat --bam [</B><I>sample.bam|sample.sam|sample.cram</I><B>] --bed [</B><I>gene_annotation.bed</I><B>]  --out </B><I>output_file</I>

[<B></B><I>OPTIONS</I>]
<A NAME="lbAD">&nbsp;</A>
<H2>DESCRIPTION</H2><P>

This mode counts the number of RNAseq reads, and the ones that overlap with an annotation file. 
We recommend using uniquely mapping reads only by specifying the correct <B>--filter-mapping-quality</B>.
<A NAME="lbAE">&nbsp;</A>
<H2>OPTIONS</H2><P>

<DL COMPACT>
<DT><B>--bed </B><I> annotation.bed</I><B></B>

<DD>
Annotation of interest
REQUIRED.
<DT><B>--bam, -b  [</B><I>in.bam</I>|<I>in.sam</I>|<I>in.cram</I><B>]</B>

<DD>
Sequence data in BAM/SAM/CRAM format.
REQUIRED.
<DT><B>--out, -o </B><I>output</I>

<DD>
Output file name
REQUIRED.
<DT><B>--filter-mapping-quality </B><I>integer</I>

<DD>
Minimum mapping quality for a read or read pair to be considered. 
Set this to only include uniquely mapped reads.
DEFAULT=10
<DT><B>--filter-keep-duplicates</B>

<DD>
Keep reads designated as duplicate by the aligner.
RECOMMENDED for RNAseq
<P>
<P>
</DL>
<A NAME="lbAF">&nbsp;</A>
<H2>OUTPUT FILE COLUMNS</H2><P>

<DL COMPACT>
<DT><B>--out</B><I> filename</I>

<DD>
This file does not have header and it contains the following columns:
<P>
<TABLE>
<TR VALIGN=top><TD ALIGN=right>1&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The total number of reads in the BAM file
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>2&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The number of mapped sequencing reads passing the <B>--filter-mapping-quality</B> 
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>3&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The number of mapped sequencing reads falling within the annotations specified with <B>--bed</B> 
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>4&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The total number of annotations in the <B>--bed</B> file
<BR></TD></TR>
<TR VALIGN=top><TD ALIGN=right>5&nbsp;&nbsp;&nbsp;&nbsp;</TD><TD>
The number of annotations covered by at least one sequencing read 
<BR></TD></TR>
</TABLE>

<P>
</DL>
<A NAME="lbAG">&nbsp;</A>
<H2>EXAMPLES</H2><P>

<DL COMPACT>
<DT>o
Running bamstat on an RNAseq sample mapped with GEM and GENCODE gene annotations:<DD>
<DT><DD>
<P>
</P><CODE style="font-size:1em;">
QTLtools bamstat --bam HG00381.chr22.bam  --out HG00381.chr22.bamstat.txt --bed gencode.v19.annotation.bed.gz --filter-mapping-quality 150 --filter-keep-duplicates
</CODE>
<P>
</DL>
<A NAME="lbAH">&nbsp;</A>
<H2>SEE ALSO</H2><P>

<I><A HREF="QTLtools.1.html">QTLtools</A></I>(1)












<A NAME="lbAI">&nbsp;</A>
<H2>BUGS</H2><P>

Please submit bugs to &lt;<A HREF="https://github.com/qtltools/qtltools">https://github.com/qtltools/qtltools</A>&gt;
<A NAME="lbAJ">&nbsp;</A>
<H2>CITATION</H2><P>

Delaneau, O., Ongen, H., Brown, A. et al. A complete tool set for molecular QTL discovery and analysis. <I>Nat Commun</I> <B>8</B>, 15452 (2017). 
&lt;<A HREF="https://doi.org/10.1038/ncomms15452">https://doi.org/10.1038/ncomms15452</A>&gt;
<A NAME="lbAK">&nbsp;</A>
<H2>AUTHORS</H2><P>

Olivier Delaneau (<A HREF="mailto:olivier.delaneau@gmail.com">olivier.delaneau@gmail.com</A>), Halit Ongen (<A HREF="mailto:halitongen@gmail.com">halitongen@gmail.com</A>)
<P>

<HR>
<A NAME="index">&nbsp;</A><H2>Index</H2><P>
<DL>
<DT><A HREF="#lbAB">NAME</A><DD>
<DT><A HREF="#lbAC">SYNOPSIS</A><DD>
<DT><A HREF="#lbAD">DESCRIPTION</A><DD>
<DT><A HREF="#lbAE">OPTIONS</A><DD>
<DT><A HREF="#lbAF">OUTPUT FILE COLUMNS</A><DD>
<DT><A HREF="#lbAG">EXAMPLES</A><DD>
<DT><A HREF="#lbAH">SEE ALSO</A><DD>
<DT><A HREF="#lbAI">BUGS</A><DD>
<DT><A HREF="#lbAJ">CITATION</A><DD>
<DT><A HREF="#lbAK">AUTHORS</A><DD>
</DL>
<HR>
This document was created by
<A HREF="/cgi-bin/man/man2html">man2html</A>,
using the manual pages.<BR>
Time: 14:11:25 GMT, August 24, 2020
</div>
</div>
</BODY>
</HTML>
